	Reorganize the alphabets/LANG/ directories, rewriting the
	sub-Makefile alphabets/LANG/Makefile.am as Makefile.am
	fragment incuded by alphabets/Makefile.am.

	Note: these include's are resolved by `automake'.

	The idea is to put everything specific for a language into
	the Makefile.am fragment for that language.

	========

	Do not distribute generated files (copied from TL2008).
	Drop unused files (from TL2008 amd more).

diff -ur -x Makefile.in -x autom4te.cache xindy.orig/configure.ac xindy/configure.ac
--- xindy.orig/configure.ac	2009-12-03 11:59:47.000000000 +0100
+++ xindy/configure.ac	2009-12-18 10:13:35.000000000 +0100
@@ -110,51 +110,6 @@
 		user-commands/Makefile \
 		make-rules/Makefile \
 		make-rules/alphabets/Makefile \
-		make-rules/alphabets/albanian/Makefile \
-		make-rules/alphabets/belarusian/Makefile \
-		make-rules/alphabets/bulgarian/Makefile \
-		make-rules/alphabets/croatian/Makefile \
-		make-rules/alphabets/czech/Makefile \
-		make-rules/alphabets/danish/Makefile \
-		make-rules/alphabets/dutch/Makefile \
-		make-rules/alphabets/english/Makefile \
-		make-rules/alphabets/esperanto/Makefile \
-		make-rules/alphabets/estonian/Makefile \
-		make-rules/alphabets/finnish/Makefile \
-		make-rules/alphabets/french/Makefile \
-		make-rules/alphabets/general/Makefile \
-		make-rules/alphabets/georgian/Makefile \
-		make-rules/alphabets/german/Makefile \
-		make-rules/alphabets/greek/Makefile \
-		make-rules/alphabets/gypsy/Makefile \
-		make-rules/alphabets/hausa/Makefile \
-		make-rules/alphabets/hebrew/Makefile \
-		make-rules/alphabets/hungarian/Makefile \
-		make-rules/alphabets/icelandic/Makefile \
-		make-rules/alphabets/italian/Makefile \
-		make-rules/alphabets/klingon/Makefile \
-		make-rules/alphabets/kurdish/Makefile \
-		make-rules/alphabets/latin/Makefile \
-		make-rules/alphabets/latvian/Makefile \
-		make-rules/alphabets/lithuanian/Makefile \
-		make-rules/alphabets/lower-sorbian/Makefile \
-		make-rules/alphabets/macedonian/Makefile \
-		make-rules/alphabets/mongolian/Makefile \
-		make-rules/alphabets/norwegian/Makefile \
-		make-rules/alphabets/polish/Makefile \
-		make-rules/alphabets/portuguese/Makefile \
-		make-rules/alphabets/romanian/Makefile \
-		make-rules/alphabets/russian/Makefile \
-		make-rules/alphabets/serbian/Makefile \
-		make-rules/alphabets/slovak/Makefile \
-		make-rules/alphabets/slovenian/Makefile \
-		make-rules/alphabets/spanish/Makefile \
-		make-rules/alphabets/swedish/Makefile \
-		make-rules/alphabets/test1/Makefile \
-		make-rules/alphabets/turkish/Makefile \
-		make-rules/alphabets/ukrainian/Makefile \
-		make-rules/alphabets/upper-sorbian/Makefile \
-		make-rules/alphabets/vietnamese/Makefile \
 		make-rules/inputenc/Makefile \
 		make-rules/styles/Makefile \
 		doc/Makefile \
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/Makefile.am xindy/make-rules/alphabets/Makefile.am
--- xindy.orig/make-rules/alphabets/Makefile.am	2009-12-02 20:34:28.000000000 +0100
+++ xindy/make-rules/alphabets/Makefile.am	2009-12-17 19:57:46.000000000 +0100
@@ -21,787 +21,86 @@
 
 langdir	= $(pkglibdir)/modules/lang
 
-DISTCLEANFILES =
+nobase_nodist_lang_DATA =
 
-albaniandir = $(langdir)/albanian
-
-ALBANIAN = \
-	albanian/latin1-lang.xdy \
-	albanian/latin1-test.xdy \
-	albanian/latin1.xdy \
-	albanian/utf8-lang.xdy \
-	albanian/utf8-test.xdy \
-	albanian/utf8.xdy
-
-nodist_albanian_DATA = $(ALBANIAN)
-
-DISTCLEANFILES += \
-	albanian/utf8.pl
-
-belarusiandir = $(langdir)/belarusian
-
-BELARUSIAN = \
-	belarusian/cp1251-lang.xdy \
-	belarusian/cp1251-test.xdy \
-	belarusian/cp1251.xdy \
-	belarusian/iso88595-lang.xdy \
-	belarusian/iso88595-test.xdy \
-	belarusian/iso88595.xdy \
-	belarusian/isoir111-lang.xdy \
-	belarusian/isoir111-test.xdy \
-	belarusian/isoir111.xdy \
-	belarusian/utf8-lang.xdy \
-	belarusian/utf8-test.xdy \
-	belarusian/utf8.xdy
-
-nodist_belarusian_DATA = $(BELARUSIAN)
-
-DISTCLEANFILES += \
-	belarusian/utf8.pl
-
-bulgariandir = $(langdir)/bulgarian
-
-BULGARIAN = \
-	bulgarian/cp1251-lang.xdy \
-	bulgarian/cp1251-test.xdy \
-	bulgarian/cp1251.xdy \
-	bulgarian/iso88595-lang.xdy \
-	bulgarian/iso88595-test.xdy \
-	bulgarian/iso88595.xdy \
-	bulgarian/koi8-r-lang.xdy \
-	bulgarian/koi8-r-test.xdy \
-	bulgarian/koi8-r.xdy \
-	bulgarian/utf8-lang.xdy \
-	bulgarian/utf8-test.xdy \
-	bulgarian/utf8.xdy
-
-nodist_bulgarian_DATA = $(BULGARIAN)
-
-DISTCLEANFILES += \
-	bulgarian/utf8.pl
-
-croatiandir = $(langdir)/croatian
-
-CROATIAN = \
-	croatian/cp1250-lang.xdy \
-	croatian/cp1250-test.xdy \
-	croatian/cp1250.xdy \
-	croatian/latin2-lang.xdy \
-	croatian/latin2-test.xdy \
-	croatian/latin2.xdy \
-	croatian/utf8-lang.xdy \
-	croatian/utf8-test.xdy \
-	croatian/utf8.xdy
-
-nodist_croatian_DATA = $(CROATIAN)
-
-DISTCLEANFILES += \
-	croatian/utf8.pl
-
-czechdir = $(langdir)/czech
-
-CZECH = \
-	czech/cp1250-lang.xdy \
-	czech/cp1250-test.xdy \
-	czech/cp1250.xdy \
-	czech/latin2-lang.xdy \
-	czech/latin2-test.xdy \
-	czech/latin2.xdy \
-	czech/utf8-lang.xdy \
-	czech/utf8-test.xdy \
-	czech/utf8.xdy
-
-nodist_czech_DATA = $(CZECH)
-
-DISTCLEANFILES += \
-	czech/utf8.pl
-
-danishdir = $(langdir)/danish
-
-DANISH = \
-	danish/cp1252-lang.xdy \
-	danish/cp1252-test.xdy \
-	danish/cp1252.xdy \
-	danish/latin9-lang.xdy \
-	danish/latin9-test.xdy \
-	danish/latin9.xdy \
-	danish/utf8-lang.xdy \
-	danish/utf8-test.xdy \
-	danish/utf8.xdy
-
-nodist_danish_DATA = $(DANISH)
-
-DISTCLEANFILES += \
-	danish/utf8.pl
-
-dutchdir = $(langdir)/dutch
-
-DUTCH = \
-	dutch/ij-as-ij-latin1-lang.xdy \
-	dutch/ij-as-ij-latin1-test.xdy \
-	dutch/ij-as-ij-latin1.xdy \
-	dutch/ij-as-ij-utf8-lang.xdy \
-	dutch/ij-as-ij-utf8-test.xdy \
-	dutch/ij-as-ij-utf8.xdy \
-	dutch/ij-as-y-utf8-lang.xdy \
-	dutch/ij-as-y-utf8-test.xdy \
-	dutch/ij-as-y-utf8.xdy
-
-nodist_dutch_DATA = $(DUTCH)
-
-DISTCLEANFILES += \
-	dutch/ij-as-ij-utf8.pl \
-	dutch/ij-as-y-utf8.pl
-
-englishdir = $(langdir)/english
-
-ENGLISH = \
-	english/cp1252-lang.xdy \
-	english/cp1252-test.xdy \
-	english/cp1252.xdy \
-	english/latin9-lang.xdy  \
-	english/latin9-test.xdy \
-	english/latin9.xdy \
-	english/utf8-lang.xdy \
-	english/utf8-test.xdy \
-	english/utf8.xdy
-
-nodist_english_DATA = $(ENGLISH)
-
-DISTCLEANFILES += \
-	english/utf8.pl
-
-esperantodir = $(langdir)/esperanto
-
-ESPERANTO = \
-	esperanto/latin3-lang.xdy \
-	esperanto/latin3-test.xdy \
-	esperanto/latin3.xdy \
-	esperanto/utf8-lang.xdy \
-	esperanto/utf8-test.xdy \
-	esperanto/utf8.xdy
-
-nodist_esperanto_DATA = $(ESPERANTO)
-
-DISTCLEANFILES += \
-	esperanto/utf8.pl
-
-estoniandir = $(langdir)/estonian
-
-ESTONIAN = \
-	estonian/cp1252-lang.xdy \
-	estonian/cp1252-test.xdy \
-	estonian/cp1252.xdy \
-	estonian/latin9-lang.xdy \
-	estonian/latin9-test.xdy \
-	estonian/latin9.xdy \
-	estonian/utf8-lang.xdy \
-	estonian/utf8-test.xdy \
-	estonian/utf8.xdy
-
-nodist_estonian_DATA = $(ESTONIAN)
-
-DISTCLEANFILES += \
-	estonian/utf8.pl
-
-finnishdir = $(langdir)/finnish
-
-FINNISH = \
-	finnish/cp1252-lang.xdy \
-	finnish/cp1252-test.xdy \
-	finnish/cp1252.xdy \
-	finnish/latin9-lang.xdy \
-	finnish/latin9-test.xdy \
-	finnish/latin9.xdy \
-	finnish/utf8-lang.xdy \
-	finnish/utf8-test.xdy \
-	finnish/utf8.xdy
-
-nodist_finnish_DATA = $(FINNISH)
-
-DISTCLEANFILES += \
-	finnish/utf8.pl
-
-frenchdir = $(langdir)/french
-
-FRENCH = \
-	french/cp1252-lang.xdy \
-	french/cp1252-test.xdy \
-	french/cp1252.xdy \
-	french/latin9-lang.xdy \
-	french/latin9-test.xdy \
-	french/latin9.xdy \
-	french/utf8-lang.xdy \
-	french/utf8-test.xdy \
-	french/utf8.xdy
-
-nodist_french_DATA = $(FRENCH)
-
-DISTCLEANFILES += \
-	french/utf8.pl
-
-generaldir = $(langdir)/general
-
-GENERAL = \
-	general/cp1252-lang.xdy \
-	general/cp1252-test.xdy \
-	general/cp1252.xdy \
-	general/cp850-lang.xdy \
-	general/cp850-test.xdy \
-	general/cp850.xdy \
-	general/latin9-lang.xdy \
-	general/latin9-test.xdy \
-	general/latin9.xdy \
-	general/utf8-lang.xdy \
-	general/utf8-test.xdy \
-	general/utf8.xdy
-
-nodist_general_DATA = $(GENERAL)
-
-DISTCLEANFILES += \
-	general/utf8.pl
-
-georgiandir = $(langdir)/georgian
-
-GEORGIAN = \
-	georgian/utf8-lang.xdy \
-	georgian/utf8-test.xdy \
-	georgian/utf8.xdy
-
-nodist_georgian_DATA = $(GEORGIAN)
-
-DISTCLEANFILES += \
-	georgian/utf8.pl
-
-germandir = $(langdir)/german
-
-GERMAN = \
-	german/braille-utf8-lang.xdy \
-	german/braille-utf8-test.xdy \
-	german/braille-utf8.xdy \
-	german/din5007-latin1-lang.xdy \
-	german/din5007-latin1-test.xdy \
-	german/din5007-latin1.xdy \
-	german/din5007-utf8-lang.xdy \
-	german/din5007-utf8-test.xdy \
-	german/din5007-utf8.xdy \
-	german/duden-latin1-lang.xdy \
-	german/duden-latin1-test.xdy \
-	german/duden-latin1.xdy \
-	german/duden-utf8-lang.xdy \
-	german/duden-utf8-test.xdy \
-	german/duden-utf8.xdy
-
-nodist_german_DATA = $(GERMAN)
-
-DISTCLEANFILES += \
-	german/braille-utf8.pl \
-	german/din5007-utf8.pl \
-	german/duden-utf8.pl
-
-greekdir = $(langdir)/greek
-
-GREEK = \
-	greek/iso88597-lang.xdy \
-	greek/iso88597-test.xdy \
-	greek/iso88597.xdy \
-	greek/polytonic-utf8-lang.xdy \
-	greek/polytonic-utf8-test.xdy \
-	greek/polytonic-utf8.xdy \
-	greek/translit-latin4-lang.xdy \
-	greek/translit-latin4-test.xdy \
-	greek/translit-latin4.xdy \
-	greek/translit-utf8-lang.xdy \
-	greek/translit-utf8-test.xdy \
-	greek/translit-utf8.xdy \
-	greek/utf8-lang.xdy \
-	greek/utf8-test.xdy \
-	greek/utf8.xdy
-
-nodist_greek_DATA = $(GREEK)
-
-DISTCLEANFILES += \
-	greek/polytonic-utf8.pl \
-	greek/translit-utf8.pl \
-	greek/utf8.pl
-
-gypsydir = $(langdir)/gypsy
-
-GYPSY = \
-	gypsy/northrussian-utf8-lang.xdy \
-	gypsy/northrussian-utf8-test.xdy \
-	gypsy/northrussian-utf8.xdy
-
-nodist_gypsy_DATA = $(GYPSY)
-
-DISTCLEANFILES += \
-	gypsy/northrussian-utf8.pl
-
-hausadir = $(langdir)/hausa
-
-HAUSA = \
-	hausa/utf8-lang.xdy \
-	hausa/utf8-test.xdy \
-	hausa/utf8.xdy
-
-nodist_hausa_DATA = $(HAUSA)
-
-DISTCLEANFILES += \
-	hausa/utf8.pl
-
-hebrewdir = $(langdir)/hebrew
-
-HEBREW = \
-	hebrew/iso88598-lang.xdy \
-	hebrew/iso88598-test.xdy \
-	hebrew/iso88598.xdy \
-	hebrew/utf8-lang.xdy \
-	hebrew/utf8-test.xdy \
-	hebrew/utf8.xdy
-
-nodist_hebrew_DATA = $(HEBREW)
-
-DISTCLEANFILES += \
-	hebrew/utf8.pl
-
-hungariandir = $(langdir)/hungarian
-
-HUNGARIAN = \
-	hungarian/cp1250-lang.xdy \
-	hungarian/cp1250-test.xdy \
-	hungarian/cp1250.xdy \
-	hungarian/latin2-lang.xdy \
-	hungarian/latin2-test.xdy \
-	hungarian/latin2.xdy \
-	hungarian/utf8-lang.xdy \
-	hungarian/utf8-test.xdy \
-	hungarian/utf8.xdy
-
-nodist_hungarian_DATA = $(HUNGARIAN)
-
-DISTCLEANFILES += \
-	hungarian/utf8.pl
-
-icelandicdir = $(langdir)/icelandic
-
-ICELANDIC = \
-	icelandic/cp1252-lang.xdy \
-	icelandic/cp1252-test.xdy \
-	icelandic/cp1252.xdy \
-	icelandic/latin9-lang.xdy \
-	icelandic/latin9-test.xdy \
-	icelandic/latin9.xdy \
-	icelandic/utf8-lang.xdy \
-	icelandic/utf8-test.xdy \
-	icelandic/utf8.xdy
-
-nodist_icelandic_DATA = $(ICELANDIC)
-
-DISTCLEANFILES += \
-	icelandic/utf8.pl
-
-italiandir = $(langdir)/italian
-
-ITALIAN = \
-	italian/latin1-lang.xdy \
-	italian/latin1-test.xdy \
-	italian/latin1.xdy \
-	italian/utf8-lang.xdy \
-	italian/utf8-test.xdy \
-	italian/utf8.xdy
-
-nodist_italian_DATA = $(ITALIAN)
-
-DISTCLEANFILES += \
-	italian/utf8.pl
-
-klingondir = $(langdir)/klingon
-
-KLINGON = \
-	klingon/utf8-lang.xdy \
-	klingon/utf8-test.xdy \
-	klingon/utf8.xdy
-
-nodist_klingon_DATA = $(KLINGON)
-
-DISTCLEANFILES += \
-	klingon/utf8.pl
-
-kurdishdir = $(langdir)/kurdish
-
-KURDISH = \
-	kurdish/bedirxan-latin5-lang.xdy \
-	kurdish/bedirxan-latin5-test.xdy \
-	kurdish/bedirxan-latin5.xdy \
-	kurdish/bedirxan-utf8-lang.xdy \
-	kurdish/bedirxan-utf8-test.xdy \
-	kurdish/bedirxan-utf8.xdy \
-	kurdish/turkish-i-latin5-lang.xdy \
-	kurdish/turkish-i-latin5-test.xdy \
-	kurdish/turkish-i-latin5.xdy \
-	kurdish/turkish-i-utf8-lang.xdy \
-	kurdish/turkish-i-utf8-test.xdy \
-	kurdish/turkish-i-utf8.xdy
-
-nodist_kurdish_DATA = $(KURDISH)
-
-DISTCLEANFILES += \
-	kurdish/bedirxan-utf8.pl \
-	kurdish/turkish-i-utf8.pl
-
-latindir = $(langdir)/latin
-
-LATIN = \
-	latin/utf8-lang.xdy \
-	latin/utf8-test.xdy \
-	latin/utf8.xdy
-
-nodist_latin_DATA = $(LATIN)
-
-DISTCLEANFILES += \
-	latin/utf8.pl
-
-latviandir = $(langdir)/latvian
-
-LATVIAN = \
-	latvian/latin4-lang.xdy \
-	latvian/latin4-test.xdy \
-	latvian/latin4.xdy \
-	latvian/utf8-lang.xdy \
-	latvian/utf8-test.xdy \
-	latvian/utf8.xdy
-
-nodist_latvian_DATA = $(LATVIAN)
-
-DISTCLEANFILES += \
-	latvian/utf8.pl
-
-lithuaniandir = $(langdir)/lithuanian
-
-LITHUANIAN = \
-	lithuanian/latin4-lang.xdy \
-	lithuanian/latin4-test.xdy \
-	lithuanian/latin4.xdy \
-	lithuanian/utf8-lang.xdy \
-	lithuanian/utf8-test.xdy \
-	lithuanian/utf8.xdy
-
-nodist_lithuanian_DATA = $(LITHUANIAN)
-
-DISTCLEANFILES += \
-	lithuanian/utf8.pl
-
-lower_sorbiandir = $(langdir)/lower-sorbian
-
-LOWER_SORBIAN = \
-	lower-sorbian/cp1250-lang.xdy \
-	lower-sorbian/cp1250-test.xdy \
-	lower-sorbian/cp1250.xdy \
-	lower-sorbian/latin2-lang.xdy \
-	lower-sorbian/latin2-test.xdy \
-	lower-sorbian/latin2.xdy \
-	lower-sorbian/utf8-lang.xdy \
-	lower-sorbian/utf8-test.xdy \
-	lower-sorbian/utf8.xdy
-
-nodist_lower_sorbian_DATA = $(LOWER_SORBIAN)
-
-DISTCLEANFILES += \
-	lower-sorbian/utf8.pl
-
-macedoniandir = $(langdir)/macedonian
-
-MACEDONIAN = \
-	macedonian/cp1251-lang.xdy \
-	macedonian/cp1251-test.xdy \
-	macedonian/cp1251.xdy \
-	macedonian/iso88595-lang.xdy \
-	macedonian/iso88595-test.xdy \
-	macedonian/iso88595.xdy \
-	macedonian/isoir111-lang.xdy \
-	macedonian/isoir111-test.xdy \
-	macedonian/isoir111.xdy \
-	macedonian/utf8-lang.xdy \
-	macedonian/utf8-test.xdy \
-	macedonian/utf8.xdy
-
-nodist_macedonian_DATA = $(MACEDONIAN)
-
-DISTCLEANFILES += \
-	macedonian/utf8.pl
-
-mongoliandir = $(langdir)/mongolian
-
-MONGOLIAN = \
-	mongolian/cyrillic-utf8-lang.xdy \
-	mongolian/cyrillic-utf8-test.xdy \
-	mongolian/cyrillic-utf8.xdy
-
-nodist_mongolian_DATA = $(MONGOLIAN)
-
-DISTCLEANFILES += \
-	mongolian/cyrillic-utf8.pl
-
-norwegiandir = $(langdir)/norwegian
-
-NORWEGIAN = \
-	norwegian/latin1-lang.xdy \
-	norwegian/latin1-test.xdy \
-	norwegian/latin1.xdy \
-	norwegian/utf8-lang.xdy \
-	norwegian/utf8-test.xdy \
-	norwegian/utf8.xdy
-
-nodist_norwegian_DATA = $(NORWEGIAN)
-
-DISTCLEANFILES += \
-	norwegian/utf8.pl
-
-polishdir = $(langdir)/polish
-
-POLISH = \
-	polish/cp1250-lang.xdy \
-	polish/cp1250-test.xdy \
-	polish/cp1250.xdy \
-	polish/latin2-lang.xdy \
-	polish/latin2-test.xdy \
-	polish/latin2.xdy \
-	polish/utf8-lang.xdy \
-	polish/utf8-test.xdy \
-	polish/utf8.xdy
-
-nodist_polish_DATA = $(POLISH)
-
-DISTCLEANFILES += \
-	polish/utf8.pl
-
-portuguesedir = $(langdir)/portuguese
-
-PORTUGUESE = \
-	portuguese/latin1-lang.xdy \
-	portuguese/latin1-test.xdy \
-	portuguese/latin1.xdy \
-	portuguese/utf8-lang.xdy \
-	portuguese/utf8-test.xdy \
-	portuguese/utf8.xdy
-
-nodist_portuguese_DATA = $(PORTUGUESE)
-
-DISTCLEANFILES += \
-	portuguese/utf8.pl
-
-romaniandir = $(langdir)/romanian
-
-ROMANIAN = \
-	romanian/latin2-lang.xdy \
-	romanian/latin2-test.xdy \
-	romanian/latin2.xdy \
-	romanian/utf8-lang.xdy \
-	romanian/utf8-test.xdy \
-	romanian/utf8.xdy
-
-nodist_romanian_DATA = $(ROMANIAN)
-
-DISTCLEANFILES += \
-	romanian/utf8.pl
-
-russiandir = $(langdir)/russian
-
-RUSSIAN = \
-	russian/cp1251-lang.xdy \
-	russian/cp1251-test.xdy \
-	russian/cp1251.xdy \
-	russian/iso88595-lang.xdy \
-	russian/iso88595-test.xdy \
-	russian/iso88595.xdy \
-	russian/koi8-r-lang.xdy \
-	russian/koi8-r-test.xdy \
-	russian/koi8-r.xdy \
-	russian/translit-iso-utf8-lang.xdy \
-	russian/translit-iso-utf8-test.xdy \
-	russian/translit-iso-utf8.xdy \
-	russian/utf8-lang.xdy \
-	russian/utf8-test.xdy \
-	russian/utf8.xdy
-
-nodist_russian_DATA = $(RUSSIAN)
-
-DISTCLEANFILES += \
-	russian/translit-iso-utf8.pl \
-	russian/utf8.pl
-
-serbiandir = $(langdir)/serbian
-
-SERBIAN = \
-	serbian/cp1251-lang.xdy \
-	serbian/cp1251-test.xdy \
-	serbian/cp1251.xdy \
-	serbian/iso88595-lang.xdy \
-	serbian/iso88595-test.xdy \
-	serbian/iso88595.xdy \
-	serbian/isoir111-lang.xdy \
-	serbian/isoir111-test.xdy \
-	serbian/isoir111.xdy \
-	serbian/utf8-lang.xdy \
-	serbian/utf8-test.xdy \
-	serbian/utf8.xdy
-
-nodist_serbian_DATA = $(SERBIAN)
-
-DISTCLEANFILES += \
-	serbian/utf8.pl
-
-slovakdir = $(langdir)/slovak
-
-SLOVAK = \
-	slovak/large-cp1250-lang.xdy \
-	slovak/large-cp1250-test.xdy \
-	slovak/large-cp1250.xdy \
-	slovak/large-latin2-lang.xdy \
-	slovak/large-latin2-test.xdy \
-	slovak/large-latin2.xdy \
-	slovak/large-utf8-lang.xdy \
-	slovak/large-utf8-test.xdy \
-	slovak/large-utf8.xdy \
-	slovak/small-cp1250-lang.xdy \
-	slovak/small-cp1250-test.xdy \
-	slovak/small-cp1250.xdy \
-	slovak/small-latin2-lang.xdy \
-	slovak/small-latin2-test.xdy \
-	slovak/small-latin2.xdy \
-	slovak/small-utf8-lang.xdy \
-	slovak/small-utf8-test.xdy \
-	slovak/small-utf8.xdy
-
-nodist_slovak_DATA = $(SLOVAK)
-
-DISTCLEANFILES += \
-	slovak/large-utf8.pl \
-	slovak/small-utf8.pl
-
-sloveniandir = $(langdir)/slovenian
-
-SLOVENIAN = \
-	slovenian/cp1250-lang.xdy \
-	slovenian/cp1250-test.xdy \
-	slovenian/cp1250.xdy \
-	slovenian/latin2-lang.xdy \
-	slovenian/latin2-test.xdy \
-	slovenian/latin2.xdy \
-	slovenian/utf8-lang.xdy \
-	slovenian/utf8-test.xdy \
-	slovenian/utf8.xdy
-
-nodist_slovenian_DATA = $(SLOVENIAN)
-
-DISTCLEANFILES += \
-	slovenian/utf8.pl
-
-spanishdir = $(langdir)/spanish
-
-SPANISH = \
-	spanish/modern-latin1-lang.xdy \
-	spanish/modern-latin1-test.xdy \
-	spanish/modern-latin1.xdy \
-	spanish/modern-utf8-lang.xdy \
-	spanish/modern-utf8-test.xdy \
-	spanish/modern-utf8.xdy \
-	spanish/traditional-latin1-lang.xdy \
-	spanish/traditional-latin1-test.xdy \
-	spanish/traditional-latin1.xdy \
-	spanish/traditional-utf8-lang.xdy \
-	spanish/traditional-utf8-test.xdy \
-	spanish/traditional-utf8.xdy
-
-nodist_spanish_DATA = $(SPANISH)
-
-DISTCLEANFILES += \
-	spanish/modern-utf8.pl \
-	spanish/traditional-utf8.pl
-
-swedishdir = $(langdir)/swedish
-
-SWEDISH = \
-	swedish/cp1252-lang.xdy \
-	swedish/cp1252-test.xdy \
-	swedish/cp1252.xdy \
-	swedish/latin9-lang.xdy \
-	swedish/latin9-test.xdy \
-	swedish/latin9.xdy \
-	swedish/utf8-lang.xdy \
-	swedish/utf8-test.xdy \
-	swedish/utf8.xdy
-
-nodist_swedish_DATA = $(SWEDISH)
-
-DISTCLEANFILES += \
-	swedish/utf8.pl
-
-turkishdir = $(langdir)/turkish
-
-TURKISH = \
-	turkish/latin3-lang.xdy \
-	turkish/latin3-test.xdy \
-	turkish/latin3.xdy \
-	turkish/latin5-lang.xdy \
-	turkish/latin5-test.xdy \
-	turkish/latin5.xdy \
-	turkish/utf8-lang.xdy \
-	turkish/utf8-test.xdy \
-	turkish/utf8.xdy
-
-nodist_turkish_DATA = $(TURKISH)
-
-DISTCLEANFILES += \
-	turkish/utf8.pl
-
-ukrainiandir = $(langdir)/ukrainian
-
-UKRAINIAN = \
-	ukrainian/cp1251-lang.xdy \
-	ukrainian/cp1251-test.xdy \
-	ukrainian/cp1251.xdy \
-	ukrainian/koi8-u-lang.xdy \
-	ukrainian/koi8-u-test.xdy \
-	ukrainian/koi8-u.xdy \
-	ukrainian/utf8-lang.xdy \
-	ukrainian/utf8-test.xdy \
-	ukrainian/utf8.xdy
-
-nodist_ukrainian_DATA = $(UKRAINIAN)
-
-DISTCLEANFILES += \
-	ukrainian/utf8.pl
-
-upper_sorbiandir = $(langdir)/upper-sorbian
-
-UPPER_SORBIAN = \
-	upper-sorbian/cp1250-lang.xdy \
-	upper-sorbian/cp1250-test.xdy \
-	upper-sorbian/cp1250.xdy \
-	upper-sorbian/latin2-lang.xdy \
-	upper-sorbian/latin2-test.xdy \
-	upper-sorbian/latin2.xdy \
-	upper-sorbian/utf8-lang.xdy \
-	upper-sorbian/utf8-test.xdy \
-	upper-sorbian/utf8.xdy
-
-nodist_upper_sorbian_DATA = $(UPPER_SORBIAN)
-
-DISTCLEANFILES += \
-	upper-sorbian/utf8.pl
-
-vietnamesedir = $(langdir)/vietnamese
-
-VIETNAMESE = \
-	vietnamese/utf8-lang.xdy \
-	vietnamese/utf8-test.xdy \
-	vietnamese/utf8.xdy
-
-nodist_vietnamese_DATA = $(VIETNAMESE)
-
-DISTCLEANFILES += \
-	vietnamese/utf8.pl
+DOCS_CP1250 =
+DOCS_CP1252 =
+DOCS_CP850 =
+DOCS_LATIN1 =
+DOCS_LATIN2 =
+DOCS_LATIN3 =
+DOCS_LATIN4 =
+DOCS_LATIN5 =
+DOCS_LATIN9 =
+
+DOCS_CYR_1251 =
+DOCS_CYR_ISO88595 = 
+DOCS_CYR_ISOIR111 =
+DOCS_CYR_KOI8_R =
+DOCS_CYR_KOI8_U =
+
+DOCS_OTHER =
+DOCS_UTF8_LAT =
+DOCS_UTF8_CYR =
+DOCS_UTF8_OTHER =
+
+EXTRA_DIST =
+CLEANFILES =
+
+subdirs =
+
+## Include all Makefile fragments.
+include albanian/Makefile.am
+include belarusian/Makefile.am
+include bulgarian/Makefile.am
+include croatian/Makefile.am
+include czech/Makefile.am
+include danish/Makefile.am
+include dutch/Makefile.am
+include english/Makefile.am
+include esperanto/Makefile.am
+include estonian/Makefile.am
+include finnish/Makefile.am
+include french/Makefile.am
+include general/Makefile.am
+include georgian/Makefile.am
+include german/Makefile.am
+include greek/Makefile.am
+include gypsy/Makefile.am
+include hausa/Makefile.am
+include hebrew/Makefile.am
+include hungarian/Makefile.am
+include icelandic/Makefile.am
+include italian/Makefile.am
+include klingon/Makefile.am
+include kurdish/Makefile.am
+include latin/Makefile.am
+include latvian/Makefile.am
+include lithuanian/Makefile.am
+include lower-sorbian/Makefile.am
+include macedonian/Makefile.am
+include mongolian/Makefile.am
+include norwegian/Makefile.am
+include polish/Makefile.am
+include portuguese/Makefile.am
+include romanian/Makefile.am
+include russian/Makefile.am
+include serbian/Makefile.am
+include slovak/Makefile.am
+include slovenian/Makefile.am
+include spanish/Makefile.am
+include swedish/Makefile.am
+include test1/Makefile.am
+include turkish/Makefile.am
+include ukrainian/Makefile.am
+include upper-sorbian/Makefile.am
+include vietnamese/Makefile.am
+
+## Make sure all language subdirectories exist.
+BUILT_SOURCES = subdirs
+subdirs:
+	@for d in $(subdirs); do $(MKDIR_P) $$d; done
+.PHONY: subdirs
 
 %utf8.pl : %utf8.pl.in
 	sed -e 's/@.{\([^}]*\)}/\1/g;' <$< >$@
@@ -814,13 +113,6 @@
 	sed -e 's/@.{\([^}]*\)}//g;' <$< | \
 	$(CONVERT_FROM_UTF8)CP850 >$@
 
-romanian/latin2.pl : romanian/utf8.pl
-	sed -e 's/@u{\([^}]*\)}//g;' -f $(srcdir)/rom-comma-kludge.sed <$< | \
-	$(CONVERT_FROM_UTF8)ISO8859-2 >$@
-
-DISTCLEANFILES += \
-	romanian/latin2.pl
-
 %latin2.pl : %utf8.pl.in
 	sed -e 's/@u{\([^}]*\)}//g;' <$< | \
 	$(CONVERT_FROM_UTF8)ISO8859-2 >$@
@@ -853,13 +145,6 @@
 	sed -e 's/@u{\([^}]*\)}//g;s/@9{\([^}]*\)}/\1/g;' <$< | \
 	$(CONVERT_FROM_UTF8)ISO-8859-15 >$@
 
-romanian/cp1250.pl : romanian/utf8.pl
-	sed -e 's/@u{\([^}]*\)}//g;' -f $(srcdir)/rom-comma-kludge.sed <$< | \
-	$(CONVERT_FROM_UTF8)CP1250 >$@
-
-DISTCLEANFILES += \
-	romanian/cp1250.pl
-
 %cp1250.pl : %utf8.pl.in
 	sed -e 's/@u{\([^}]*\)}//g;' <$< | \
 	$(CONVERT_FROM_UTF8)CP1250 >$@
@@ -911,203 +196,11 @@
 	  fi; \
 	fi
 
-SUBDIRS = \
-	albanian \
-	belarusian \
-	bulgarian \
-	croatian \
-	czech \
-	danish \
-	dutch \
-	english \
-	esperanto \
-	estonian \
-	finnish \
-	french \
-	general \
-	georgian \
-	german \
-	greek \
-	gypsy \
-	hausa \
-	hebrew \
-	hungarian \
-	icelandic \
-	italian \
-	klingon \
-	kurdish \
-	latin \
-	latvian \
-	lithuanian \
-	lower-sorbian \
-	macedonian \
-	mongolian \
-	norwegian \
-	polish \
-	portuguese \
-	romanian \
-	russian \
-	serbian \
-	slovak \
-	slovenian \
-	spanish \
-	swedish \
-	test1 \
-	turkish \
-	ukrainian \
-	upper-sorbian \
-	vietnamese
-
 SCRS = \
 	alphabets-doc.tex \
 	make-rules.pl \
 	utf8-to-tex.sed
 
-ALPHABETS = \
-	$(ALBANIAN) \
-	$(BELARUSIAN) \
-	$(BULGARIAN) \
-	$(CROATIAN) \
-	$(CZECH) \
-	$(DANISH) \
-	$(DUTCH) \
-	$(ENGLISH) \
-	$(ESPERANTO) \
-	$(ESTONIAN) \
-	$(FINNISH) \
-	$(FRENCH) \
-	$(GENERAL) \
-	$(GEORGIAN) \
-	$(GERMAN) \
-	$(GREEK) \
-	$(GYPSY) \
-	$(HAUSA) \
-	$(HEBREW) \
-	$(HUNGARIAN) \
-	$(ICELANDIC) \
-	$(ITALIAN) \
-	$(KLINGON) \
-	$(KURDISH) \
-	$(LATIN) \
-	$(LATVIAN) \
-	$(LITHUANIAN) \
-	$(LOWER_SORBIAN) \
-	$(MACEDONIAN) \
-	$(MONGOLIAN) \
-	$(NORWEGIAN) \
-	$(POLISH) \
-	$(PORTUGUESE) \
-	$(ROMANIAN) \
-	$(RUSSIAN) \
-	$(SERBIAN) \
-	$(SLOVAK) \
-	$(SLOVENIAN) \
-	$(SPANISH) \
-	$(SWEDISH) \
-	$(TURKISH) \
-	$(UKRAINIAN) \
-	$(UPPER_SORBIAN) \
-	$(VIETNAMESE)
-
-DOCS_CP1250 = \
-	croatian/cp1250-doc.tex \
-	czech/cp1250-doc.tex \
-	hungarian/cp1250-doc.tex \
-	lower-sorbian/cp1250-doc.tex \
-	polish/cp1250-doc.tex \
-	slovak/large-cp1250-doc.tex \
-	slovak/small-cp1250-doc.tex \
-	slovenian/cp1250-doc.tex \
-	upper-sorbian/cp1250-doc.tex
-
-DOCS_CP1252 = \
-	danish/cp1252-doc.tex \
-	english/cp1252-doc.tex \
-	estonian/cp1252-doc.tex \
-	finnish/cp1252-doc.tex \
-	french/cp1252-doc.tex \
-	general/cp1252-doc.tex \
-	icelandic/cp1252-doc.tex \
-	swedish/cp1252-doc.tex
-
-DOCS_CP850 = \
-	general/cp850-doc.tex
-
-DOCS_LATIN1 = \
-	albanian/latin1-doc.tex \
-	dutch/ij-as-ij-latin1-doc.tex \
-	german/din5007-latin1-doc.tex \
-	german/duden-latin1-doc.tex \
-	italian/latin1-doc.tex \
-	norwegian/latin1-doc.tex \
-	portuguese/latin1-doc.tex \
-	spanish/modern-latin1-doc.tex \
-	spanish/traditional-latin1-doc.tex
-
-
-DOCS_LATIN2 = \
-	croatian/latin2-doc.tex \
-	czech/latin2-doc.tex \
-	hungarian/latin2-doc.tex \
-	lower-sorbian/latin2-doc.tex \
-	polish/latin2-doc.tex \
-	romanian/latin2-doc.tex \
-	slovak/small-latin2-doc.tex \
-	slovak/large-latin2-doc.tex \
-	slovenian/latin2-doc.tex \
-	upper-sorbian/latin2-doc.tex
-
-DOCS_LATIN3 = \
-	esperanto/latin3-doc.tex \
-	turkish/latin3-doc.tex
-
-DOCS_LATIN4 = \
-	greek/translit-latin4-doc.tex \
-	latvian/latin4-doc.tex \
-	lithuanian/latin4-doc.tex
-
-DOCS_LATIN5 = \
-	kurdish/bedirxan-latin5-doc.tex \
-	kurdish/turkish-i-latin5-doc.tex \
-	turkish/latin5-doc.tex
-
-DOCS_LATIN9 = \
-	danish/latin9-doc.tex \
-	english/latin9-doc.tex \
-	estonian/latin9-doc.tex \
-	finnish/latin9-doc.tex \
-	french/latin9-doc.tex \
-	general/latin9-doc.tex \
-	icelandic/latin9-doc.tex \
-	swedish/latin9-doc.tex
-
-DOCS_CYR_1251 = \
-	belarusian/cp1251-doc.tex \
-	bulgarian/cp1251-doc.tex \
-	macedonian/cp1251-doc.tex \
-	russian/cp1251-doc.tex \
-	serbian/cp1251-doc.tex \
-	ukrainian/cp1251-doc.tex
-
-DOCS_CYR_ISO88595 = \
-	belarusian/iso88595-doc.tex \
-	bulgarian/iso88595-doc.tex \
-	macedonian/iso88595-doc.tex \
-	russian/iso88595-doc.tex \
-	serbian/iso88595-doc.tex
-
-DOCS_CYR_ISOIR111 = \
-	belarusian/isoir111-doc.tex \
-	macedonian/isoir111-doc.tex \
-	serbian/isoir111-doc.tex
-
-DOCS_CYR_KOI8_R = \
-	bulgarian/koi8-r-doc.tex \
-	russian/koi8-r-doc.tex
-
-DOCS_CYR_KOI8_U = \
-	ukrainian/koi8-u-doc.tex
-
 DOCS_CYR = \
 	$(DOCS_CYR_1251) \
 	$(DOCS_CYR_ISO88595) \
@@ -1115,72 +208,6 @@
 	$(DOCS_CYR_KOI8_R) \
 	$(DOCS_CYR_KOI8_U)
 
-
-DOCS_OTHER = \
-	greek/iso88597-doc.tex \
-	hebrew/iso88598-doc.tex
-
-
-DOCS_UTF8_LAT = \
-	albanian/utf8-doc.tex \
-	croatian/utf8-doc.tex \
-	czech/utf8-doc.tex \
-	danish/utf8-doc.tex \
-	dutch/ij-as-ij-utf8-doc.tex \
-	dutch/ij-as-y-utf8-doc.tex \
-	english/utf8-doc.tex \
-	esperanto/utf8-doc.tex \
-	estonian/utf8-doc.tex \
-	finnish/utf8-doc.tex \
-	french/utf8-doc.tex \
-	general/utf8-doc.tex \
-	german/din5007-utf8-doc.tex \
-	german/duden-utf8-doc.tex \
-	greek/translit-utf8-doc.tex \
-	greek/utf8-doc.tex \
-	gypsy/northrussian-utf8-doc.tex \
-	hausa/utf8-doc.tex \
-	hungarian/utf8-doc.tex \
-	icelandic/utf8-doc.tex \
-	italian/utf8-doc.tex \
-	kurdish/bedirxan-utf8-doc.tex \
-	kurdish/turkish-i-utf8-doc.tex \
-	latin/utf8-doc.tex \
-	latvian/utf8-doc.tex \
-	lithuanian/utf8-doc.tex \
-	lower-sorbian/utf8-doc.tex \
-	norwegian/utf8-doc.tex \
-	polish/utf8-doc.tex \
-	portuguese/utf8-doc.tex \
-	romanian/utf8-doc.tex \
-	russian/translit-iso-utf8-doc.tex \
-	slovak/small-utf8-doc.tex \
-	slovak/large-utf8-doc.tex \
-	slovenian/utf8-doc.tex \
-	spanish/modern-utf8-doc.tex \
-	spanish/traditional-utf8-doc.tex \
-	swedish/utf8-doc.tex \
-	turkish/utf8-doc.tex \
-	upper-sorbian/utf8-doc.tex \
-	vietnamese/utf8-doc.tex
-
-DOCS_UTF8_CYR = \
-	belarusian/utf8-doc.tex \
-	bulgarian/utf8-doc.tex \
-	macedonian/utf8-doc.tex \
-	mongolian/cyrillic-utf8-doc.tex \
-	russian/utf8-doc.tex \
-	serbian/utf8-doc.tex \
-	ukrainian/utf8-doc.tex
-
-DOCS_UTF8_OTHER = \
-	georgian/utf8-doc.tex \
-	german/braille-utf8-doc.tex \
-	greek/utf8-doc.tex \
-	greek/polytonic-utf8-doc.tex \
-	hebrew/utf8-doc.tex \
-	klingon/utf8-doc.tex
-
 DOCS_CPS = \
 	$(DOCS_CP1250) \
 	$(DOCS_CP1252) \
@@ -1270,7 +297,7 @@
 	alphabets-inc.tex \
 	alphabets-doc.toc
 
-EXTRA_DIST = $(SCRS) test.xdy rom-comma-kludge.sed
+EXTRA_DIST += $(SCRS) test.xdy
 
-CLEANFILES = $(ALPHABETS) $(DOC_CLEAN)
+CLEANFILES += $(nobase_nodist_lang_DATA) $(DOCS) $(DOC_CLEAN)
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/albanian/Makefile.am xindy/make-rules/alphabets/albanian/Makefile.am
--- xindy.orig/make-rules/alphabets/albanian/Makefile.am	2008-01-09 15:01:31.000000000 +0100
+++ xindy/make-rules/alphabets/albanian/Makefile.am	2009-12-17 20:41:39.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for albanian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,22 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin1.pl utf8.pl.in
+subdirs += albanian
 
-CLEANFILES = \
-	latin1.raw \
-	latin1-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	albanian/latin1-lang.xdy albanian/latin1-test.xdy albanian/latin1.xdy \
+	albanian/utf8-lang.xdy albanian/utf8-test.xdy albanian/utf8.xdy
 
+DOCS_LATIN1 += \
+	albanian/latin1-doc.tex
 
+DOCS_UTF8_LAT += \
+	albanian/utf8-doc.tex
 
+EXTRA_DIST += \
+	albanian/utf8.pl.in
+
+CLEANFILES += \
+	albanian/latin1.pl albanian/latin1.raw \
+	albanian/utf8.pl albanian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/belarusian/Makefile.am xindy/make-rules/alphabets/belarusian/Makefile.am
--- xindy.orig/make-rules/alphabets/belarusian/Makefile.am	2008-01-09 15:01:30.000000000 +0100
+++ xindy/make-rules/alphabets/belarusian/Makefile.am	2009-12-17 20:42:46.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for belarusian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,19 +18,32 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1251.pl iso88595.pl isoir111.pl utf8.pl.in
-
-CLEANFILES = \
-	cp1251.raw \
-	cp1251-doc.tex \
-	iso88595.raw \
-	iso88595-doc.tex \
-	isoir111.raw \
-	isoir111-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
-
-
-
+subdirs += belarusian
 
+nobase_nodist_lang_DATA += \
+	belarusian/cp1251-lang.xdy belarusian/cp1251-test.xdy belarusian/cp1251.xdy \
+	belarusian/iso88595-lang.xdy belarusian/iso88595-test.xdy belarusian/iso88595.xdy \
+	belarusian/isoir111-lang.xdy belarusian/isoir111-test.xdy belarusian/isoir111.xdy \
+	belarusian/utf8-lang.xdy belarusian/utf8-test.xdy belarusian/utf8.xdy
+
+DOCS_CYR_1251 += \
+	belarusian/cp1251-doc.tex
+
+DOCS_CYR_ISO88595 += \
+	belarusian/iso88595-doc.tex
+
+DOCS_CYR_ISOIR111 += \
+	belarusian/isoir111-doc.tex
+
+DOCS_UTF8_CYR += \
+	belarusian/utf8-doc.tex
+
+EXTRA_DIST += \
+	belarusian/utf8.pl.in
+
+CLEANFILES += \
+	belarusian/cp1251.pl belarusian/cp1251.raw \
+	belarusian/iso88595.pl belarusian/iso88595.raw \
+	belarusian/isoir111.pl belarusian/isoir111.raw \
+	belarusian/utf8.pl belarusian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/bulgarian/Makefile.am xindy/make-rules/alphabets/bulgarian/Makefile.am
--- xindy.orig/make-rules/alphabets/bulgarian/Makefile.am	2008-01-09 15:01:30.000000000 +0100
+++ xindy/make-rules/alphabets/bulgarian/Makefile.am	2009-12-17 20:43:40.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for bulgarian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,19 +18,33 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1251.pl iso88595.pl koi8-r.pl utf8.pl.in
+subdirs += bulgarian
 
+nobase_nodist_lang_DATA += \
+	bulgarian/cp1251-lang.xdy bulgarian/cp1251-test.xdy bulgarian/cp1251.xdy \
+	bulgarian/iso88595-lang.xdy bulgarian/iso88595-test.xdy bulgarian/iso88595.xdy \
+	bulgarian/koi8-r-lang.xdy bulgarian/koi8-r-test.xdy bulgarian/koi8-r.xdy \
+	bulgarian/utf8-lang.xdy bulgarian/utf8-test.xdy bulgarian/utf8.xdy
 
-CLEANFILES = \
-	cp1251.raw \
-	cp1251-doc.tex \
-	iso88595.raw \
-	iso88595-doc.tex \
-	koi8-r.raw \
-	koi8-r-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+DOCS_CYR_1251 += \
+	bulgarian/cp1251-doc.tex
 
+DOCS_CYR_ISO88595 += \
+	bulgarian/iso88595-doc.tex
 
+DOCS_CYR_KOI8_R += \
+	bulgarian/koi8-r-doc.tex
 
+DOCS_UTF8_CYR += \
+	bulgarian/utf8-doc.tex
+
+EXTRA_DIST += \
+	bulgarian/utf8.pl.in
+
+
+CLEANFILES += \
+	bulgarian/cp1251.pl bulgarian/cp1251.raw \
+	bulgarian/iso88595.pl bulgarian/iso88595.raw \
+	bulgarian/koi8-r.pl bulgarian/koi8-r.raw \
+	bulgarian/utf8.pl bulgarian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/croatian/Makefile.am xindy/make-rules/alphabets/croatian/Makefile.am
--- xindy.orig/make-rules/alphabets/croatian/Makefile.am	2008-01-09 15:01:29.000000000 +0100
+++ xindy/make-rules/alphabets/croatian/Makefile.am	2009-12-17 20:44:31.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for croatian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1250.pl latin2.pl utf8.pl.in
+subdirs += croatian
 
-CLEANFILES = \
-	cp1250.raw \
-	cp1250-doc.tex \
-	latin2.raw \
-	latin2-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	croatian/cp1250-lang.xdy croatian/cp1250-test.xdy croatian/cp1250.xdy \
+	croatian/latin2-lang.xdy croatian/latin2-test.xdy croatian/latin2.xdy \
+	croatian/utf8-lang.xdy croatian/utf8-test.xdy croatian/utf8.xdy
 
+DOCS_CP1250 += \
+	croatian/cp1250-doc.tex
 
+DOCS_LATIN2 += \
+	croatian/latin2-doc.tex
 
+DOCS_UTF8_LAT += \
+	croatian/utf8-doc.tex
+
+EXTRA_DIST += \
+	croatian/utf8.pl.in
+
+CLEANFILES += \
+	croatian/cp1250.pl croatian/cp1250.raw \
+	croatian/latin2.pl croatian/latin2.raw \
+	croatian/utf8.pl croatian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/czech/Makefile.am xindy/make-rules/alphabets/czech/Makefile.am
--- xindy.orig/make-rules/alphabets/czech/Makefile.am	2008-01-09 15:01:29.000000000 +0100
+++ xindy/make-rules/alphabets/czech/Makefile.am	2009-12-17 20:45:08.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for czech.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1250.pl latin2.pl utf8.pl.in
+subdirs += czech
 
-CLEANFILES = \
-	cp1250.raw \
-	cp1250-doc.tex \
-	latin2.raw \
-	latin2-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	czech/cp1250-lang.xdy czech/cp1250-test.xdy czech/cp1250.xdy \
+	czech/latin2-lang.xdy czech/latin2-test.xdy czech/latin2.xdy \
+	czech/utf8-lang.xdy czech/utf8-test.xdy czech/utf8.xdy
 
+DOCS_CP1250 += \
+	czech/cp1250-doc.tex
 
+DOCS_LATIN2 += \
+	czech/latin2-doc.tex
 
+DOCS_UTF8_LAT += \
+	czech/utf8-doc.tex
+
+EXTRA_DIST += \
+	czech/utf8.pl.in
+
+CLEANFILES += \
+	czech/cp1250.pl czech/cp1250.raw \
+	czech/latin2.pl czech/latin2.raw \
+	czech/utf8.pl czech/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/danish/Makefile.am xindy/make-rules/alphabets/danish/Makefile.am
--- xindy.orig/make-rules/alphabets/danish/Makefile.am	2008-01-09 15:01:28.000000000 +0100
+++ xindy/make-rules/alphabets/danish/Makefile.am	2009-12-17 20:45:50.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for danish.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,19 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-    cp1252.pl latin9.pl utf8.pl.in \
-    latin1.xdy
+subdirs += danish
 
+nobase_nodist_lang_DATA += \
+	danish/cp1252-lang.xdy danish/cp1252-test.xdy danish/cp1252.xdy \
+	danish/latin9-lang.xdy danish/latin9-test.xdy danish/latin9.xdy \
+	danish/utf8-lang.xdy danish/utf8-test.xdy danish/utf8.xdy
 
-CLEANFILES = \
-	cp1252.raw \
-	cp1252-doc.tex \
-	latin9.raw \
-	latin9-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+DOCS_CP1252 += \
+	danish/cp1252-doc.tex
 
+DOCS_LATIN9 += \
+	danish/latin9-doc.tex
 
+DOCS_UTF8_LAT += \
+	danish/utf8-doc.tex
 
+EXTRA_DIST += \
+	danish/utf8.pl.in
+
+CLEANFILES += \
+	danish/cp1252.pl danish/cp1252.raw \
+	danish/latin9.pl danish/latin9.raw \
+	danish/utf8.pl danish/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/dutch/Makefile.am xindy/make-rules/alphabets/dutch/Makefile.am
--- xindy.orig/make-rules/alphabets/dutch/Makefile.am	2008-01-09 15:01:27.000000000 +0100
+++ xindy/make-rules/alphabets/dutch/Makefile.am	2009-12-17 20:47:48.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for dutch.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,18 +18,26 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	ij-as-ij-latin1.pl ij-as-y-utf8.pl.in ij-as-ij-utf8.pl.in \
-	latin1.xdy tex-latin1.xdy
-
-CLEANFILES = \
-	ij-as-ij-latin1.raw \
-	ij-as-ij-latin1-doc.tex \
-	ij-as-ij-utf8.raw \
-	ij-as-ij-utf8-doc.tex \
-	ij-as-y-utf8.raw \
-	ij-as-y-utf8-doc.tex
-
-
+subdirs += dutch
 
+nobase_nodist_lang_DATA += \
+	dutch/ij-as-ij-latin1-lang.xdy dutch/ij-as-ij-latin1-test.xdy dutch/ij-as-ij-latin1.xdy \
+	dutch/ij-as-ij-utf8-lang.xdy dutch/ij-as-ij-utf8-test.xdy dutch/ij-as-ij-utf8.xdy \
+	dutch/ij-as-y-utf8-lang.xdy dutch/ij-as-y-utf8-test.xdy dutch/ij-as-y-utf8.xdy
+
+DOCS_LATIN1 += \
+	dutch/ij-as-ij-latin1-doc.tex
+
+DOCS_UTF8_LAT += \
+	dutch/ij-as-ij-utf8-doc.tex \
+	dutch/ij-as-y-utf8-doc.tex
+
+EXTRA_DIST += \
+	dutch/ij-as-ij-utf8.pl.in \
+	dutch/ij-as-y-utf8.pl.in
+
+CLEANFILES += \
+	dutch/ij-as-ij-latin1.pl dutch/ij-as-ij-latin1.raw \
+	dutch/ij-as-ij-utf8.pl dutch/ij-as-ij-utf8.raw \
+	dutch/ij-as-y-utf8.pl dutch/ij-as-y-utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/english/Makefile.am xindy/make-rules/alphabets/english/Makefile.am
--- xindy.orig/make-rules/alphabets/english/Makefile.am	2008-01-09 15:01:27.000000000 +0100
+++ xindy/make-rules/alphabets/english/Makefile.am	2009-12-17 20:48:26.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for english.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,18 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	cp1252.pl latin9.pl  utf8.pl.in \
-	ascii.xdy tex-ascii.xdy
-
-CLEANFILES = \
-	cp1252.raw \
-	cp1252-doc.tex \
-	latin9.raw  \
-	latin9-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+subdirs += english
 
+nobase_nodist_lang_DATA += \
+	english/cp1252-lang.xdy english/cp1252-test.xdy english/cp1252.xdy \
+	english/latin9-lang.xdy  english/latin9-test.xdy english/latin9.xdy \
+	english/utf8-lang.xdy english/utf8-test.xdy english/utf8.xdy
 
+DOCS_CP1252 += \
+	english/cp1252-doc.tex
 
+DOCS_UTF8_LAT += \
+	english/utf8-doc.tex
+
+DOCS_LATIN9 += \
+	english/latin9-doc.tex
+
+EXTRA_DIST += \
+	english/utf8.pl.in
+
+CLEANFILES += \
+	english/cp1252.pl english/cp1252.raw \
+	english/latin9.pl english/latin9.raw  \
+	english/utf8.pl english/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/esperanto/Makefile.am xindy/make-rules/alphabets/esperanto/Makefile.am
--- xindy.orig/make-rules/alphabets/esperanto/Makefile.am	2008-01-09 15:01:26.000000000 +0100
+++ xindy/make-rules/alphabets/esperanto/Makefile.am	2009-12-17 20:48:53.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for esperanto.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,22 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin3.pl utf8.pl.in
+subdirs += esperanto
 
-CLEANFILES = \
-	latin3.raw \
-	latin3-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	esperanto/latin3-lang.xdy esperanto/latin3-test.xdy esperanto/latin3.xdy \
+	esperanto/utf8-lang.xdy esperanto/utf8-test.xdy esperanto/utf8.xdy
 
+DOCS_LATIN3 += \
+	esperanto/latin3-doc.tex
 
+DOCS_UTF8_LAT += \
+	esperanto/utf8-doc.tex
 
+EXTRA_DIST += \
+	esperanto/utf8.pl.in
+
+CLEANFILES += \
+	esperanto/latin3.pl esperanto/latin3.raw \
+	esperanto/utf8.pl esperanto/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/estonian/Makefile.am xindy/make-rules/alphabets/estonian/Makefile.am
--- xindy.orig/make-rules/alphabets/estonian/Makefile.am	2008-01-09 15:01:26.000000000 +0100
+++ xindy/make-rules/alphabets/estonian/Makefile.am	2009-12-17 20:49:33.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for estonian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1252.pl latin9.pl utf8.pl.in
+subdirs += estonian
 
-CLEANFILES = \
-	cp1252.raw \
-	cp1252-doc.tex \
-	latin9.raw \
-	latin9-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	estonian/cp1252-lang.xdy estonian/cp1252-test.xdy estonian/cp1252.xdy \
+	estonian/latin9-lang.xdy estonian/latin9-test.xdy estonian/latin9.xdy \
+	estonian/utf8-lang.xdy estonian/utf8-test.xdy estonian/utf8.xdy
 
+DOCS_CP1252 += \
+	estonian/cp1252-doc.tex
 
+DOCS_LATIN9 += \
+	estonian/latin9-doc.tex
 
+DOCS_UTF8_LAT += \
+	estonian/utf8-doc.tex
+
+EXTRA_DIST += \
+	estonian/utf8.pl.in
+
+CLEANFILES += \
+	estonian/cp1252.pl estonian/cp1252.raw \
+	estonian/latin9.pl estonian/latin9.raw \
+	estonian/utf8.pl estonian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/finnish/Makefile.am xindy/make-rules/alphabets/finnish/Makefile.am
--- xindy.orig/make-rules/alphabets/finnish/Makefile.am	2008-01-09 15:01:26.000000000 +0100
+++ xindy/make-rules/alphabets/finnish/Makefile.am	2009-12-17 20:52:37.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for finnish.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,19 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	cp1252.pl latin9.pl utf8.pl.in \
-	tex-cp1252.xdy  tex-latin9.xdy
+subdirs += finnish
 
+nobase_nodist_lang_DATA += \
+	finnish/cp1252-lang.xdy finnish/cp1252-test.xdy finnish/cp1252.xdy \
+	finnish/latin9-lang.xdy finnish/latin9-test.xdy finnish/latin9.xdy \
+	finnish/utf8-lang.xdy finnish/utf8-test.xdy finnish/utf8.xdy
 
-CLEANFILES = \
-	cp1252.raw \
-	cp1252-doc.tex \
-	latin9.raw \
-	latin9-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+DOCS_CP1252 += \
+	finnish/cp1252-doc.tex
 
+DOCS_LATIN9 += \
+	finnish/latin9-doc.tex
 
+DOCS_UTF8_LAT += \
+	finnish/utf8-doc.tex
 
+EXTRA_DIST += \
+	finnish/utf8.pl.in
+
+CLEANFILES += \
+	finnish/cp1252.pl finnish/cp1252.raw \
+	finnish/latin9.pl finnish/latin9.raw \
+	finnish/utf8.pl finnish/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/french/Makefile.am xindy/make-rules/alphabets/french/Makefile.am
--- xindy.orig/make-rules/alphabets/french/Makefile.am	2008-01-09 15:01:25.000000000 +0100
+++ xindy/make-rules/alphabets/french/Makefile.am	2009-12-17 20:53:56.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for french.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1252.pl latin9.pl utf8.pl.in
+subdirs += french
 
-CLEANFILES = \
-	cp1252.raw \
-	cp1252-doc.tex \
-	latin9.raw \
-	latin9-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	french/cp1252-lang.xdy french/cp1252-test.xdy french/cp1252.xdy \
+	french/latin9-lang.xdy french/latin9-test.xdy french/latin9.xdy \
+	french/utf8-lang.xdy french/utf8-test.xdy french/utf8.xdy
 
+DOCS_CP1252 += \
+	french/cp1252-doc.tex
 
+DOCS_LATIN9 += \
+	french/latin9-doc.tex
 
+DOCS_UTF8_LAT += \
+	french/utf8-doc.tex
+
+EXTRA_DIST += \
+	french/utf8.pl.in
+
+CLEANFILES += \
+	french/cp1252.pl french/cp1252.raw \
+	french/latin9.pl french/latin9.raw \
+	french/utf8.pl french/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/general/Makefile.am xindy/make-rules/alphabets/general/Makefile.am
--- xindy.orig/make-rules/alphabets/general/Makefile.am	2008-01-09 15:01:25.000000000 +0100
+++ xindy/make-rules/alphabets/general/Makefile.am	2009-12-17 20:55:12.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for general.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,20 +18,33 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	cp1252.pl cp850.pl latin9.pl template.pl utf8.pl.in \
-	latin1.xdy
-
-CLEANFILES = \
-	cp1252.raw \
-	cp1252-doc.tex \
-	cp850.raw \
-	cp850-doc.tex \
-	latin9.raw \
-	latin9-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
-
-
+subdirs += general
 
+nobase_nodist_lang_DATA += \
+	general/cp1252-lang.xdy general/cp1252-test.xdy general/cp1252.xdy \
+	general/cp850-lang.xdy general/cp850-test.xdy general/cp850.xdy \
+	general/latin9-lang.xdy general/latin9-test.xdy general/latin9.xdy \
+	general/utf8-lang.xdy general/utf8-test.xdy general/utf8.xdy
+
+DOCS_CP1252 += \
+	general/cp1252-doc.tex
+
+DOCS_CP850 += \
+	general/cp850-doc.tex
+
+DOCS_LATIN9 += \
+	general/latin9-doc.tex
+
+DOCS_UTF8_LAT += \
+	general/utf8-doc.tex
+
+EXTRA_DIST += \
+	general/template.pl \
+	general/utf8.pl.in
+
+CLEANFILES += \
+	general/cp1252.pl general/cp1252.raw \
+	general/cp850.pl general/cp850.raw \
+	general/latin9.pl general/latin9.raw \
+	general/utf8.pl general/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/georgian/Makefile.am xindy/make-rules/alphabets/georgian/Makefile.am
--- xindy.orig/make-rules/alphabets/georgian/Makefile.am	2008-01-09 15:01:24.000000000 +0100
+++ xindy/make-rules/alphabets/georgian/Makefile.am	2009-12-17 20:55:26.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for .
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,12 +18,17 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = utf8.pl.in
+subdirs += georgian
 
-CLEANFILES = \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	georgian/utf8-lang.xdy georgian/utf8-test.xdy georgian/utf8.xdy
 
+DOCS_UTF8_OTHER += \
+	georgian/utf8-doc.tex
 
+EXTRA_DIST += \
+	georgian/utf8.pl.in
 
+CLEANFILES += \
+	georgian/utf8.pl georgian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/german/Makefile.am xindy/make-rules/alphabets/german/Makefile.am
--- xindy.orig/make-rules/alphabets/german/Makefile.am	2008-01-09 15:01:22.000000000 +0100
+++ xindy/make-rules/alphabets/german/Makefile.am	2009-12-17 22:37:57.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for german.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,23 +18,35 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	braille-utf8.pl.in  duden-latin1.pl \
-	din5007-latin1.pl duden-utf8.pl.in din5007-utf8.pl.in \
-	tex-din5007-latin1.xdy tex-duden-latin1.xdy tex-latin1.xdy
-
-CLEANFILES = \
-	braille-utf8.raw \
-	braille-utf8-doc.tex \
-	din5007-latin1.raw \
-	din5007-latin1-doc.tex \
-	din5007-utf8.raw \
-	din5007-utf8-doc.tex \
-	duden-latin1.raw \
-	duden-latin1-doc.tex \
-	duden-utf8.raw \
-	duden-utf8-doc.tex
-
-
+subdirs += german
 
+nobase_nodist_lang_DATA += \
+	german/braille-utf8-lang.xdy german/braille-utf8-test.xdy german/braille-utf8.xdy \
+	german/din5007-latin1-lang.xdy german/din5007-latin1-test.xdy german/din5007-latin1.xdy \
+	german/din5007-utf8-lang.xdy german/din5007-utf8-test.xdy german/din5007-utf8.xdy \
+	german/duden-latin1-lang.xdy german/duden-latin1-test.xdy german/duden-latin1.xdy \
+	german/duden-utf8-lang.xdy german/duden-utf8-test.xdy german/duden-utf8.xdy
+
+DOCS_LATIN1 += \
+	german/din5007-latin1-doc.tex \
+	german/duden-latin1-doc.tex
+
+DOCS_UTF8_LAT += \
+	german/din5007-utf8-doc.tex \
+	german/duden-utf8-doc.tex
+
+DOCS_UTF8_OTHER += \
+	german/braille-utf8-doc.tex
+
+EXTRA_DIST += \
+	german/braille-utf8.pl.in \
+	german/duden-utf8.pl.in \
+	german/din5007-utf8.pl.in
+
+CLEANFILES += \
+	german/braille-utf8.pl german/braille-utf8.raw \
+	german/din5007-latin1.pl german/din5007-latin1.raw \
+	german/din5007-utf8.pl german/din5007-utf8.raw \
+	german/duden-latin1.pl german/duden-latin1.raw \
+	german/duden-utf8.pl german/duden-utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/greek/Makefile.am xindy/make-rules/alphabets/greek/Makefile.am
--- xindy.orig/make-rules/alphabets/greek/Makefile.am	2008-01-09 15:01:22.000000000 +0100
+++ xindy/make-rules/alphabets/greek/Makefile.am	2009-12-17 21:02:59.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for greek.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,24 +18,39 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	iso88597.pl  polytonic-utf8.pl.in utf8.pl.in \
-	translit-latin4.pl translit-utf8.pl.in \
-	translit-tex.xdy
-
-
-CLEANFILES = \
-	iso88597.raw \
-	iso88597-doc.tex \
-	polytonic-utf8.raw \
-	polytonic-utf8-doc.tex \
-	translit-latin4.raw \
-	translit-latin4-doc.tex \
-	translit-utf8.raw \
-	translit-utf8-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
-
-
+subdirs += greek
 
+nobase_nodist_lang_DATA += \
+	greek/iso88597-lang.xdy greek/iso88597-test.xdy greek/iso88597.xdy \
+	greek/polytonic-utf8-lang.xdy greek/polytonic-utf8-test.xdy greek/polytonic-utf8.xdy \
+	greek/translit-latin4-lang.xdy greek/translit-latin4-test.xdy greek/translit-latin4.xdy \
+	greek/translit-utf8-lang.xdy greek/translit-utf8-test.xdy greek/translit-utf8.xdy \
+	greek/utf8-lang.xdy greek/utf8-test.xdy greek/utf8.xdy
+
+DOCS_LATIN4 += \
+	greek/translit-latin4-doc.tex
+
+DOCS_OTHER += \
+	greek/iso88597-doc.tex
+
+DOCS_UTF8_LAT += \
+	greek/translit-utf8-doc.tex \
+	greek/utf8-doc.tex
+
+DOCS_UTF8_OTHER += \
+	greek/utf8-doc.tex \
+	greek/polytonic-utf8-doc.tex
+
+EXTRA_DIST += \
+	greek/polytonic-utf8.pl.in \
+	greek/translit-utf8.pl.in \
+	greek/utf8.pl.in
+
+
+CLEANFILES += \
+	greek/iso88597.pl greek/iso88597.raw \
+	greek/polytonic-utf8.pl greek/polytonic-utf8.raw \
+	greek/translit-latin4.pl greek/translit-latin4.raw \
+	greek/translit-utf8.pl greek/translit-utf8.raw \
+	greek/utf8.pl greek/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/gypsy/Makefile.am xindy/make-rules/alphabets/gypsy/Makefile.am
--- xindy.orig/make-rules/alphabets/gypsy/Makefile.am	2008-01-09 15:01:21.000000000 +0100
+++ xindy/make-rules/alphabets/gypsy/Makefile.am	2009-12-17 21:03:13.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for gypsy.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,11 +18,17 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = northrussian-utf8.pl.in
+subdirs += gypsy
 
-CLEANFILES = \
-	northrussian-utf8.raw \
-	northrussian-utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	gypsy/northrussian-utf8-lang.xdy gypsy/northrussian-utf8-test.xdy gypsy/northrussian-utf8.xdy
 
+DOCS_UTF8_LAT += \
+	gypsy/northrussian-utf8-doc.tex
 
+EXTRA_DIST += \
+	gypsy/northrussian-utf8.pl.in
+
+CLEANFILES += \
+	gypsy/northrussian-utf8.pl gypsy/northrussian-utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/hausa/Makefile.am xindy/make-rules/alphabets/hausa/Makefile.am
--- xindy.orig/make-rules/alphabets/hausa/Makefile.am	2008-01-09 15:01:21.000000000 +0100
+++ xindy/make-rules/alphabets/hausa/Makefile.am	2009-12-17 21:03:24.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for hausa.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,12 +18,17 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = utf8.pl.in
+subdirs += hausa
 
-CLEANFILES = \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	hausa/utf8-lang.xdy hausa/utf8-test.xdy hausa/utf8.xdy
 
+DOCS_UTF8_LAT += \
+	hausa/utf8-doc.tex
 
+EXTRA_DIST += \
+	hausa/utf8.pl.in
 
+CLEANFILES += \
+	hausa/utf8.pl hausa/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/hebrew/Makefile.am xindy/make-rules/alphabets/hebrew/Makefile.am
--- xindy.orig/make-rules/alphabets/hebrew/Makefile.am	2008-01-09 15:01:21.000000000 +0100
+++ xindy/make-rules/alphabets/hebrew/Makefile.am	2009-12-17 21:03:49.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for hebrew.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,22 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = iso88598.pl utf8.pl.in
+subdirs += hebrew
 
-CLEANFILES = \
-	iso88598.raw \
-	iso88598-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	hebrew/iso88598-lang.xdy hebrew/iso88598-test.xdy hebrew/iso88598.xdy \
+	hebrew/utf8-lang.xdy hebrew/utf8-test.xdy hebrew/utf8.xdy
 
+DOCS_OTHER += \
+	hebrew/iso88598-doc.tex
 
+DOCS_UTF8_OTHER += \
+	hebrew/utf8-doc.tex
 
+EXTRA_DIST += \
+	hebrew/utf8.pl.in
+
+CLEANFILES += \
+	hebrew/iso88598.pl hebrew/iso88598.raw \
+	hebrew/utf8.pl hebrew/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/hungarian/Makefile.am xindy/make-rules/alphabets/hungarian/Makefile.am
--- xindy.orig/make-rules/alphabets/hungarian/Makefile.am	2008-01-09 15:01:20.000000000 +0100
+++ xindy/make-rules/alphabets/hungarian/Makefile.am	2009-12-17 21:05:21.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for hungarian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1250.pl latin2.pl utf8.pl.in
+subdirs += hungarian
 
-CLEANFILES = \
-	cp1250.raw \
-	cp1250-doc.tex \
-	latin2.raw \
-	latin2-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	hungarian/cp1250-lang.xdy hungarian/cp1250-test.xdy hungarian/cp1250.xdy \
+	hungarian/latin2-lang.xdy hungarian/latin2-test.xdy hungarian/latin2.xdy \
+	hungarian/utf8-lang.xdy hungarian/utf8-test.xdy hungarian/utf8.xdy
 
+DOCS_CP1250 += \
+	hungarian/cp1250-doc.tex
 
+DOCS_LATIN2 += \
+	hungarian/latin2-doc.tex
 
+DOCS_UTF8_LAT += \
+	hungarian/utf8-doc.tex
+
+EXTRA_DIST += \
+	hungarian/utf8.pl.in
+
+CLEANFILES += \
+	hungarian/cp1250.pl hungarian/cp1250.raw \
+	hungarian/latin2.pl hungarian/latin2.raw \
+	hungarian/utf8.pl hungarian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/icelandic/Makefile.am xindy/make-rules/alphabets/icelandic/Makefile.am
--- xindy.orig/make-rules/alphabets/icelandic/Makefile.am	2008-01-09 15:01:20.000000000 +0100
+++ xindy/make-rules/alphabets/icelandic/Makefile.am	2009-12-17 21:05:55.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for icelandic.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,19 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	cp1252.pl latin9.pl utf8.pl.in \
-	latin1.xdy
+subdirs += icelandic
 
+nobase_nodist_lang_DATA += \
+	icelandic/cp1252-lang.xdy icelandic/cp1252-test.xdy icelandic/cp1252.xdy \
+	icelandic/latin9-lang.xdy icelandic/latin9-test.xdy icelandic/latin9.xdy \
+	icelandic/utf8-lang.xdy icelandic/utf8-test.xdy icelandic/utf8.xdy
 
-CLEANFILES = \
-	cp1252.raw \
-	cp1252-doc.tex \
-	latin9.raw \
-	latin9-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+DOCS_CP1252 += \
+	icelandic/cp1252-doc.tex
 
+DOCS_LATIN9 += \
+	icelandic/latin9-doc.tex
 
+DOCS_UTF8_LAT += \
+	icelandic/utf8-doc.tex
 
+EXTRA_DIST += \
+	icelandic/utf8.pl.in
+
+CLEANFILES += \
+	icelandic/cp1252.pl icelandic/cp1252.raw \
+	icelandic/latin9.pl icelandic/latin9.raw \
+	icelandic/utf8.pl icelandic/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/italian/Makefile.am xindy/make-rules/alphabets/italian/Makefile.am
--- xindy.orig/make-rules/alphabets/italian/Makefile.am	2008-01-09 15:01:19.000000000 +0100
+++ xindy/make-rules/alphabets/italian/Makefile.am	2009-12-17 21:06:18.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for italian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,22 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin1.pl utf8.pl.in
+subdirs += italian
 
-CLEANFILES = \
-	latin1.raw \
-	latin1-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	italian/latin1-lang.xdy italian/latin1-test.xdy italian/latin1.xdy \
+	italian/utf8-lang.xdy italian/utf8-test.xdy italian/utf8.xdy
 
+DOCS_LATIN1 += \
+	italian/latin1-doc.tex
 
+DOCS_UTF8_LAT += \
+	italian/utf8-doc.tex
 
+EXTRA_DIST += \
+	italian/utf8.pl.in
+
+CLEANFILES += \
+	italian/latin1.pl italian/latin1.raw \
+	italian/utf8.pl italian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/klingon/Makefile.am xindy/make-rules/alphabets/klingon/Makefile.am
--- xindy.orig/make-rules/alphabets/klingon/Makefile.am	2008-01-09 15:01:19.000000000 +0100
+++ xindy/make-rules/alphabets/klingon/Makefile.am	2009-12-17 21:28:22.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for klingon.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,15 +18,21 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	utf8.pl.in \
-	translit-ascii.pl
+subdirs += klingon
 
+nobase_nodist_lang_DATA += \
+	klingon/utf8-lang.xdy klingon/utf8-test.xdy klingon/utf8.xdy
 
-CLEANFILES = \
-	utf8.raw \
-	utf8-doc.tex
+DOCS_UTF8_OTHER += \
+	klingon/utf8-doc.tex
 
+EXTRA_DIST += \
+	klingon/utf8.pl.in
 
+EXTRA_DIST += \
+	klingon/translit-ascii.pl
 
 
+CLEANFILES += \
+	klingon/utf8.pl klingon/utf8.raw
+
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/kurdish/Makefile.am xindy/make-rules/alphabets/kurdish/Makefile.am
--- xindy.orig/make-rules/alphabets/kurdish/Makefile.am	2008-01-09 15:01:18.000000000 +0100
+++ xindy/make-rules/alphabets/kurdish/Makefile.am	2009-12-17 21:07:54.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for kurdish.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,20 +18,29 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-	bedirxan-latin5.pl bedirxan-utf8.pl.in \
-	turkish-i-latin5.pl turkish-i-utf8.pl.in \
-	latin5.xdy
-
-CLEANFILES = \
-	bedirxan-latin5.raw \
-	bedirxan-latin5-doc.tex \
-	bedirxan-utf8.raw \
-	bedirxan-utf8-doc.tex \
-	turkish-i-latin5.raw \
-	turkish-i-latin5-doc.tex \
-	turkish-i-utf8.raw \
-	turkish-i-utf8-doc.tex
-
+subdirs += kurdish
 
+nobase_nodist_lang_DATA += \
+	kurdish/bedirxan-latin5-lang.xdy kurdish/bedirxan-latin5-test.xdy kurdish/bedirxan-latin5.xdy \
+	kurdish/bedirxan-utf8-lang.xdy kurdish/bedirxan-utf8-test.xdy kurdish/bedirxan-utf8.xdy \
+	kurdish/turkish-i-latin5-lang.xdy kurdish/turkish-i-latin5-test.xdy kurdish/turkish-i-latin5.xdy \
+	kurdish/turkish-i-utf8-lang.xdy kurdish/turkish-i-utf8-test.xdy kurdish/turkish-i-utf8.xdy
+
+DOCS_LATIN5 += \
+	kurdish/bedirxan-latin5-doc.tex \
+	kurdish/turkish-i-latin5-doc.tex
+
+DOCS_UTF8_LAT += \
+	kurdish/bedirxan-utf8-doc.tex \
+	kurdish/turkish-i-utf8-doc.tex
+
+EXTRA_DIST += \
+	kurdish/bedirxan-utf8.pl.in \
+	kurdish/turkish-i-utf8.pl.in
+
+CLEANFILES += \
+	kurdish/bedirxan-latin5.pl kurdish/bedirxan-latin5.raw \
+	kurdish/bedirxan-utf8.pl kurdish/bedirxan-utf8.raw \
+	kurdish/turkish-i-latin5.pl kurdish/turkish-i-latin5.raw \
+	kurdish/turkish-i-utf8.pl kurdish/turkish-i-utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/latin/Makefile.am xindy/make-rules/alphabets/latin/Makefile.am
--- xindy.orig/make-rules/alphabets/latin/Makefile.am	2008-01-09 15:01:18.000000000 +0100
+++ xindy/make-rules/alphabets/latin/Makefile.am	2009-12-17 21:08:11.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for latin.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,12 +18,17 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = utf8.pl.in
+subdirs += latin
 
-CLEANFILES = \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	latin/utf8-lang.xdy latin/utf8-test.xdy latin/utf8.xdy
 
+DOCS_UTF8_LAT += \
+	latin/utf8-doc.tex
 
+EXTRA_DIST += \
+	latin/utf8.pl.in
 
+CLEANFILES += \
+	latin/utf8.pl latin/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/latvian/Makefile.am xindy/make-rules/alphabets/latvian/Makefile.am
--- xindy.orig/make-rules/alphabets/latvian/Makefile.am	2008-01-09 15:01:17.000000000 +0100
+++ xindy/make-rules/alphabets/latvian/Makefile.am	2009-12-17 21:08:35.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for latvian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,22 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin4.pl utf8.pl.in
+subdirs += latvian
 
-CLEANFILES = \
-	latin4.raw \
-	latin4-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	latvian/latin4-lang.xdy latvian/latin4-test.xdy latvian/latin4.xdy \
+	latvian/utf8-lang.xdy latvian/utf8-test.xdy latvian/utf8.xdy
 
+DOCS_LATIN4 += \
+	latvian/latin4-doc.tex
 
+DOCS_UTF8_LAT += \
+	latvian/utf8-doc.tex
 
+EXTRA_DIST += \
+	latvian/utf8.pl.in
+
+CLEANFILES += \
+	latvian/latin4.pl latvian/latin4.raw \
+	latvian/utf8.pl latvian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/lithuanian/Makefile.am xindy/make-rules/alphabets/lithuanian/Makefile.am
--- xindy.orig/make-rules/alphabets/lithuanian/Makefile.am	2008-01-09 15:01:17.000000000 +0100
+++ xindy/make-rules/alphabets/lithuanian/Makefile.am	2009-12-17 21:09:06.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for lithuanian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,22 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin4.pl utf8.pl.in
+subdirs += lithuanian
 
-CLEANFILES = \
-	latin4.raw \
-	latin4-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	lithuanian/latin4-lang.xdy lithuanian/latin4-test.xdy lithuanian/latin4.xdy \
+	lithuanian/utf8-lang.xdy lithuanian/utf8-test.xdy lithuanian/utf8.xdy
 
+DOCS_LATIN4 += \
+	lithuanian/latin4-doc.tex
 
+DOCS_UTF8_LAT += \
+	lithuanian/utf8-doc.tex
 
+EXTRA_DIST += \
+	lithuanian/utf8.pl.in
+
+CLEANFILES += \
+	lithuanian/latin4.pl lithuanian/latin4.raw \
+	lithuanian/utf8.pl lithuanian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/lower-sorbian/Makefile.am xindy/make-rules/alphabets/lower-sorbian/Makefile.am
--- xindy.orig/make-rules/alphabets/lower-sorbian/Makefile.am	2008-01-09 15:01:16.000000000 +0100
+++ xindy/make-rules/alphabets/lower-sorbian/Makefile.am	2009-12-17 21:09:55.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for .
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1250.pl latin2.pl utf8.pl.in
+subdirs += lower-sorbian
 
-CLEANFILES = \
-	cp1250.raw \
-	cp1250-doc.tex \
-	latin2.raw \
-	latin2-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	lower-sorbian/cp1250-lang.xdy lower-sorbian/cp1250-test.xdy lower-sorbian/cp1250.xdy \
+	lower-sorbian/latin2-lang.xdy lower-sorbian/latin2-test.xdy lower-sorbian/latin2.xdy \
+	lower-sorbian/utf8-lang.xdy lower-sorbian/utf8-test.xdy lower-sorbian/utf8.xdy
 
+DOCS_CP1250 += \
+	lower-sorbian/cp1250-doc.tex
 
+DOCS_LATIN2 += \
+	lower-sorbian/latin2-doc.tex
 
+DOCS_UTF8_LAT += \
+	lower-sorbian/utf8-doc.tex
+
+EXTRA_DIST += \
+	lower-sorbian/utf8.pl.in
+
+CLEANFILES += \
+	lower-sorbian/cp1250.pl lower-sorbian/cp1250.raw \
+	lower-sorbian/latin2.pl lower-sorbian/latin2.raw \
+	lower-sorbian/utf8.pl lower-sorbian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/macedonian/Makefile.am xindy/make-rules/alphabets/macedonian/Makefile.am
--- xindy.orig/make-rules/alphabets/macedonian/Makefile.am	2008-01-09 15:01:16.000000000 +0100
+++ xindy/make-rules/alphabets/macedonian/Makefile.am	2009-12-17 21:10:51.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for macedonian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,18 +18,32 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1251.pl iso88595.pl isoir111.pl utf8.pl.in
-
-CLEANFILES = \
-	cp1251.raw \
-	cp1251-doc.tex \
-	iso88595.raw \
-	iso88595-doc.tex \
-	isoir111.raw \
-	isoir111-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
-
-
+subdirs += macedonian
 
+nobase_nodist_lang_DATA += \
+	macedonian/cp1251-lang.xdy macedonian/cp1251-test.xdy macedonian/cp1251.xdy \
+	macedonian/iso88595-lang.xdy macedonian/iso88595-test.xdy macedonian/iso88595.xdy \
+	macedonian/isoir111-lang.xdy macedonian/isoir111-test.xdy macedonian/isoir111.xdy \
+	macedonian/utf8-lang.xdy macedonian/utf8-test.xdy macedonian/utf8.xdy
+
+DOCS_CYR_1251 += \
+	macedonian/cp1251-doc.tex
+
+DOCS_CYR_ISO88595 += \
+	macedonian/iso88595-doc.tex
+
+DOCS_CYR_ISOIR111 += \
+	macedonian/isoir111-doc.tex
+
+DOCS_UTF8_CYR += \
+	macedonian/utf8-doc.tex
+
+EXTRA_DIST += \
+	macedonian/utf8.pl.in
+
+CLEANFILES += \
+	macedonian/cp1251.pl macedonian/cp1251.raw \
+	macedonian/iso88595.pl macedonian/iso88595.raw \
+	macedonian/isoir111.pl macedonian/isoir111.raw \
+	macedonian/utf8.pl macedonian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/mongolian/Makefile.am xindy/make-rules/alphabets/mongolian/Makefile.am
--- xindy.orig/make-rules/alphabets/mongolian/Makefile.am	2008-01-09 15:01:15.000000000 +0100
+++ xindy/make-rules/alphabets/mongolian/Makefile.am	2009-12-17 21:11:05.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for mongolian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,12 +18,17 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cyrillic-utf8.pl.in
+subdirs += mongolian
 
-CLEANFILES = \
-	cyrillic-utf8.raw \
-	cyrillic-utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	mongolian/cyrillic-utf8-lang.xdy mongolian/cyrillic-utf8-test.xdy mongolian/cyrillic-utf8.xdy
 
+DOCS_UTF8_CYR += \
+	mongolian/cyrillic-utf8-doc.tex
 
+EXTRA_DIST += \
+	mongolian/cyrillic-utf8.pl.in
 
+CLEANFILES += \
+	mongolian/cyrillic-utf8.pl mongolian/cyrillic-utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/norwegian/Makefile.am xindy/make-rules/alphabets/norwegian/Makefile.am
--- xindy.orig/make-rules/alphabets/norwegian/Makefile.am	2008-01-09 15:01:15.000000000 +0100
+++ xindy/make-rules/alphabets/norwegian/Makefile.am	2009-12-17 21:11:29.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for norwegian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,22 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin1.pl utf8.pl.in
+subdirs += norwegian
 
-CLEANFILES = \
-	latin1.raw \
-	latin1-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	norwegian/latin1-lang.xdy norwegian/latin1-test.xdy norwegian/latin1.xdy \
+	norwegian/utf8-lang.xdy norwegian/utf8-test.xdy norwegian/utf8.xdy
 
+DOCS_LATIN1 += \
+	norwegian/latin1-doc.tex
 
+DOCS_UTF8_LAT += \
+	norwegian/utf8-doc.tex
 
+EXTRA_DIST += \
+	norwegian/utf8.pl.in
+
+CLEANFILES += \
+	norwegian/latin1.pl norwegian/latin1.raw \
+	norwegian/utf8.pl norwegian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/polish/Makefile.am xindy/make-rules/alphabets/polish/Makefile.am
--- xindy.orig/make-rules/alphabets/polish/Makefile.am	2008-01-09 15:01:14.000000000 +0100
+++ xindy/make-rules/alphabets/polish/Makefile.am	2009-12-17 21:12:03.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for polish.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1250.pl latin2.pl utf8.pl.in
+subdirs += polish
 
-CLEANFILES = \
-	cp1250.raw \
-	cp1250-doc.tex \
-	latin2.raw \
-	latin2-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	polish/cp1250-lang.xdy polish/cp1250-test.xdy polish/cp1250.xdy \
+	polish/latin2-lang.xdy polish/latin2-test.xdy polish/latin2.xdy \
+	polish/utf8-lang.xdy polish/utf8-test.xdy polish/utf8.xdy
 
+DOCS_CP1250 += \
+	polish/cp1250-doc.tex
 
+DOCS_LATIN2 += \
+	polish/latin2-doc.tex
 
+DOCS_UTF8_LAT += \
+	polish/utf8-doc.tex
+
+EXTRA_DIST += \
+	polish/utf8.pl.in
+
+CLEANFILES += \
+	polish/cp1250.pl polish/cp1250.raw \
+	polish/latin2.pl polish/latin2.raw \
+	polish/utf8.pl polish/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/portuguese/Makefile.am xindy/make-rules/alphabets/portuguese/Makefile.am
--- xindy.orig/make-rules/alphabets/portuguese/Makefile.am	2008-01-09 15:01:14.000000000 +0100
+++ xindy/make-rules/alphabets/portuguese/Makefile.am	2009-12-17 21:12:27.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for portuguese.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,22 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin1.pl utf8.pl.in
+subdirs += portuguese
 
-CLEANFILES = \
-	latin1.raw \
-	latin1-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	portuguese/latin1-lang.xdy portuguese/latin1-test.xdy portuguese/latin1.xdy \
+	portuguese/utf8-lang.xdy portuguese/utf8-test.xdy portuguese/utf8.xdy
 
+DOCS_LATIN1 += \
+	portuguese/latin1-doc.tex
 
+DOCS_UTF8_LAT += \
+	portuguese/utf8-doc.tex
 
+EXTRA_DIST += \
+	portuguese/utf8.pl.in
+
+CLEANFILES += \
+	portuguese/latin1.pl portuguese/latin1.raw \
+	portuguese/utf8.pl portuguese/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/romanian/Makefile.am xindy/make-rules/alphabets/romanian/Makefile.am
--- xindy.orig/make-rules/alphabets/romanian/Makefile.am	2008-01-09 15:01:14.000000000 +0100
+++ xindy/make-rules/alphabets/romanian/Makefile.am	2009-12-17 21:16:41.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for romanian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,14 +18,32 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin2.pl utf8.pl.in
-
-CLEANFILES = \
-	latin2.raw \
-	latin2-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
-
-
+subdirs += romanian
 
+nobase_nodist_lang_DATA += \
+	romanian/latin2-lang.xdy romanian/latin2-test.xdy romanian/latin2.xdy \
+	romanian/utf8-lang.xdy romanian/utf8-test.xdy romanian/utf8.xdy
+
+DOCS_LATIN2 += \
+	romanian/latin2-doc.tex
+
+DOCS_UTF8_LAT += \
+	romanian/utf8-doc.tex
+
+romanian/cp1250.pl : romanian/utf8.pl
+	sed -e 's/@u{\([^}]*\)}//g;' -f $(srcdir)/rom-comma-kludge.sed <$< | \
+	$(CONVERT_FROM_UTF8)CP1250 >$@
+
+romanian/latin2.pl : romanian/utf8.pl
+	sed -e 's/@u{\([^}]*\)}//g;' -f $(srcdir)/rom-comma-kludge.sed <$< | \
+	$(CONVERT_FROM_UTF8)ISO8859-2 >$@
+
+EXTRA_DIST += \
+	rom-comma-kludge.sed \
+	romanian/utf8.pl.in
+
+CLEANFILES += \
+	romanian/cp1250.raw romanian/cp1250.pl \
+	romanian/latin2.pl romanian/latin2.raw \
+	romanian/utf8.pl romanian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/russian/Makefile.am xindy/make-rules/alphabets/russian/Makefile.am
--- xindy.orig/make-rules/alphabets/russian/Makefile.am	2008-01-09 15:01:13.000000000 +0100
+++ xindy/make-rules/alphabets/russian/Makefile.am	2009-12-17 21:18:03.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for russian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,22 +18,38 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-    cp1251.pl iso88595.pl koi8-r.pl translit-iso-utf8.pl.in utf8.pl.in \
-    translit-iso-tex.xdy translit-tex.xdy translit-tgl-tex.xdy
-
-CLEANFILES = \
-	cp1251.raw \
-	cp1251-doc.tex \
-	iso88595.raw \
-	iso88595-doc.tex \
-	koi8-r.raw \
-	koi8-r-doc.tex \
-	translit-iso-utf8.raw \
-	translit-iso-utf8-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
-
-
+subdirs += russian
 
+nobase_nodist_lang_DATA += \
+	russian/cp1251-lang.xdy russian/cp1251-test.xdy russian/cp1251.xdy \
+	russian/iso88595-lang.xdy russian/iso88595-test.xdy russian/iso88595.xdy \
+	russian/koi8-r-lang.xdy russian/koi8-r-test.xdy russian/koi8-r.xdy \
+	russian/translit-iso-utf8-lang.xdy russian/translit-iso-utf8-test.xdy russian/translit-iso-utf8.xdy \
+	russian/utf8-lang.xdy russian/utf8-test.xdy russian/utf8.xdy
+
+DOCS_CYR_1251 += \
+	russian/cp1251-doc.tex
+
+DOCS_CYR_ISO88595 += \
+	russian/iso88595-doc.tex
+
+DOCS_CYR_KOI8_R += \
+	russian/koi8-r-doc.tex
+
+DOCS_UTF8_LAT += \
+	russian/translit-iso-utf8-doc.tex
+
+DOCS_UTF8_CYR += \
+	russian/utf8-doc.tex
+
+EXTRA_DIST += \
+	russian/translit-iso-utf8.pl.in \
+	russian/utf8.pl.in
+
+CLEANFILES += \
+	russian/cp1251.pl russian/cp1251.raw \
+	russian/iso88595.pl russian/iso88595.raw \
+	russian/koi8-r.pl russian/koi8-r.raw \
+	russian/translit-iso-utf8.pl russian/translit-iso-utf8.raw \
+	russian/utf8.pl russian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/serbian/Makefile.am xindy/make-rules/alphabets/serbian/Makefile.am
--- xindy.orig/make-rules/alphabets/serbian/Makefile.am	2008-01-09 15:01:13.000000000 +0100
+++ xindy/make-rules/alphabets/serbian/Makefile.am	2009-12-17 21:18:57.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for serbian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,18 +18,32 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1251.pl iso88595.pl isoir111.pl utf8.pl.in
-
-CLEANFILES = \
-	cp1251.raw \
-	cp1251-doc.tex \
-	iso88595.raw \
-	iso88595-doc.tex \
-	isoir111.raw \
-	isoir111-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
-
-
+subdirs += serbian
 
+nobase_nodist_lang_DATA += \
+	serbian/cp1251-lang.xdy serbian/cp1251-test.xdy serbian/cp1251.xdy \
+	serbian/iso88595-lang.xdy serbian/iso88595-test.xdy serbian/iso88595.xdy \
+	serbian/isoir111-lang.xdy serbian/isoir111-test.xdy serbian/isoir111.xdy \
+	serbian/utf8-lang.xdy serbian/utf8-test.xdy serbian/utf8.xdy
+
+DOCS_CYR_1251 += \
+	serbian/cp1251-doc.tex
+
+DOCS_CYR_ISO88595 += \
+	serbian/iso88595-doc.tex
+
+DOCS_CYR_ISOIR111 += \
+	serbian/isoir111-doc.tex
+
+DOCS_UTF8_CYR += \
+	serbian/utf8-doc.tex
+
+EXTRA_DIST += \
+	serbian/utf8.pl.in
+
+CLEANFILES += \
+	serbian/cp1251.pl serbian/cp1251.raw \
+	serbian/iso88595.pl serbian/iso88595.raw \
+	serbian/isoir111.pl serbian/isoir111.raw \
+	serbian/utf8.pl serbian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/slovak/Makefile.am xindy/make-rules/alphabets/slovak/Makefile.am
--- xindy.orig/make-rules/alphabets/slovak/Makefile.am	2008-01-09 15:01:12.000000000 +0100
+++ xindy/make-rules/alphabets/slovak/Makefile.am	2009-12-17 21:20:23.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for slovak.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,24 +18,37 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-    large-cp1250.pl large-latin2.pl large-utf8.pl.in \
-    small-cp1250.pl small-latin2.pl small-utf8.pl.in
-
-CLEANFILES = \
-	large-cp1250.raw \
-	large-cp1250-doc.tex \
-	large-latin2.raw \
-	large-latin2-doc.tex \
-	large-utf8.raw \
-	large-utf8-doc.tex \
-	small-cp1250.raw \
-	small-cp1250-doc.tex \
-	small-latin2.raw \
-	small-latin2-doc.tex \
-	small-utf8.raw \
-	small-utf8-doc.tex
-
-
+subdirs += slovak
 
+nobase_nodist_lang_DATA += \
+	slovak/large-cp1250-lang.xdy slovak/large-cp1250-test.xdy slovak/large-cp1250.xdy \
+	slovak/large-latin2-lang.xdy slovak/large-latin2-test.xdy slovak/large-latin2.xdy \
+	slovak/large-utf8-lang.xdy slovak/large-utf8-test.xdy slovak/large-utf8.xdy \
+	slovak/small-cp1250-lang.xdy slovak/small-cp1250-test.xdy slovak/small-cp1250.xdy \
+	slovak/small-latin2-lang.xdy slovak/small-latin2-test.xdy slovak/small-latin2.xdy \
+	slovak/small-utf8-lang.xdy slovak/small-utf8-test.xdy slovak/small-utf8.xdy
+
+DOCS_CP1250 += \
+	slovak/large-cp1250-doc.tex \
+	slovak/small-cp1250-doc.tex
+
+DOCS_LATIN2 += \
+	slovak/small-latin2-doc.tex \
+	slovak/large-latin2-doc.tex
+
+DOCS_UTF8_LAT += \
+	slovak/small-utf8-doc.tex \
+	slovak/large-utf8-doc.tex
+
+EXTRA_DIST += \
+	slovak/large-utf8.pl.in \
+	slovak/small-utf8.pl.in
+
+CLEANFILES += \
+	slovak/large-cp1250.pl slovak/large-cp1250.raw \
+	slovak/large-latin2.pl slovak/large-latin2.raw \
+	slovak/large-utf8.pl slovak/large-utf8.raw \
+	slovak/small-cp1250.pl slovak/small-cp1250.raw \
+	slovak/small-latin2.pl slovak/small-latin2.raw \
+	slovak/small-utf8.pl slovak/small-utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/slovenian/Makefile.am xindy/make-rules/alphabets/slovenian/Makefile.am
--- xindy.orig/make-rules/alphabets/slovenian/Makefile.am	2008-01-09 15:01:12.000000000 +0100
+++ xindy/make-rules/alphabets/slovenian/Makefile.am	2009-12-17 21:20:57.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for slovenian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1250.pl latin2.pl utf8.pl.in
+subdirs += slovenian
 
-CLEANFILES = \
-	cp1250.raw \
-	cp1250-doc.tex \
-	latin2.raw \
-	latin2-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	slovenian/cp1250-lang.xdy slovenian/cp1250-test.xdy slovenian/cp1250.xdy \
+	slovenian/latin2-lang.xdy slovenian/latin2-test.xdy slovenian/latin2.xdy \
+	slovenian/utf8-lang.xdy slovenian/utf8-test.xdy slovenian/utf8.xdy
 
+DOCS_CP1250 += \
+	slovenian/cp1250-doc.tex
 
+DOCS_LATIN2 += \
+	slovenian/latin2-doc.tex
 
+DOCS_UTF8_LAT += \
+	slovenian/utf8-doc.tex
+
+EXTRA_DIST += \
+	slovenian/utf8.pl.in
+
+CLEANFILES += \
+	slovenian/cp1250.pl slovenian/cp1250.raw \
+	slovenian/latin2.pl slovenian/latin2.raw \
+	slovenian/utf8.pl slovenian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/spanish/Makefile.am xindy/make-rules/alphabets/spanish/Makefile.am
--- xindy.orig/make-rules/alphabets/spanish/Makefile.am	2008-01-09 15:01:11.000000000 +0100
+++ xindy/make-rules/alphabets/spanish/Makefile.am	2009-12-17 21:21:50.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for spanish.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,20 +18,29 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-    modern-latin1.pl modern-utf8.pl.in \
-    traditional-latin1.pl traditional-utf8.pl.in
-
-CLEANFILES = \
-	modern-latin1.raw \
-	modern-latin1-doc.tex \
-	modern-utf8.raw \
-	modern-utf8-doc.tex \
-	traditional-latin1.raw \
-	traditional-latin1-doc.tex \
-	traditional-utf8.raw \
-	traditional-utf8-doc.tex
-
-
+subdirs += spanish
 
+nobase_nodist_lang_DATA += \
+	spanish/modern-latin1-lang.xdy spanish/modern-latin1-test.xdy spanish/modern-latin1.xdy \
+	spanish/modern-utf8-lang.xdy spanish/modern-utf8-test.xdy spanish/modern-utf8.xdy \
+	spanish/traditional-latin1-lang.xdy spanish/traditional-latin1-test.xdy spanish/traditional-latin1.xdy \
+	spanish/traditional-utf8-lang.xdy spanish/traditional-utf8-test.xdy spanish/traditional-utf8.xdy
+
+DOCS_LATIN1 += \
+	spanish/modern-latin1-doc.tex \
+	spanish/traditional-latin1-doc.tex
+
+DOCS_UTF8_LAT += \
+	spanish/modern-utf8-doc.tex \
+	spanish/traditional-utf8-doc.tex
+
+EXTRA_DIST += \
+	spanish/modern-utf8.pl.in \
+	spanish/traditional-utf8.pl.in
+
+CLEANFILES += \
+	spanish/modern-latin1.pl spanish/modern-latin1.raw \
+	spanish/modern-utf8.pl spanish/modern-utf8.raw \
+	spanish/traditional-latin1.pl spanish/traditional-latin1.raw \
+	spanish/traditional-utf8.pl spanish/traditional-utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/swedish/Makefile.am xindy/make-rules/alphabets/swedish/Makefile.am
--- xindy.orig/make-rules/alphabets/swedish/Makefile.am	2008-01-09 15:01:11.000000000 +0100
+++ xindy/make-rules/alphabets/swedish/Makefile.am	2009-12-17 21:22:22.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for swedish.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,18 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = \
-    cp1252.pl latin9.pl utf8.pl.in \
-    latin1.xdy
-
-CLEANFILES = \
-	cp1252.raw \
-	cp1252-doc.tex \
-	latin9.raw \
-	latin9-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+subdirs += swedish
 
+nobase_nodist_lang_DATA += \
+	swedish/cp1252-lang.xdy swedish/cp1252-test.xdy swedish/cp1252.xdy \
+	swedish/latin9-lang.xdy swedish/latin9-test.xdy swedish/latin9.xdy \
+	swedish/utf8-lang.xdy swedish/utf8-test.xdy swedish/utf8.xdy
 
+DOCS_CP1252 += \
+	swedish/cp1252-doc.tex
 
+DOCS_LATIN9 += \
+	swedish/latin9-doc.tex
+
+DOCS_UTF8_LAT += \
+	swedish/utf8-doc.tex
+
+EXTRA_DIST += \
+	swedish/utf8.pl.in
+
+CLEANFILES += \
+	swedish/cp1252.pl swedish/cp1252.raw \
+	swedish/latin9.pl swedish/latin9.raw \
+	swedish/utf8.pl swedish/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/test1/Makefile.am xindy/make-rules/alphabets/test1/Makefile.am
--- xindy.orig/make-rules/alphabets/test1/Makefile.am	2008-01-09 15:01:11.000000000 +0100
+++ xindy/make-rules/alphabets/test1/Makefile.am	2009-12-04 01:23:47.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for test1.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,10 +18,8 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = test1.pl
-
-
-
-
+subdirs += test1
 
+EXTRA_DIST += \
+	test1/test1.pl
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/turkish/Makefile.am xindy/make-rules/alphabets/turkish/Makefile.am
--- xindy.orig/make-rules/alphabets/turkish/Makefile.am	2008-01-09 15:01:10.000000000 +0100
+++ xindy/make-rules/alphabets/turkish/Makefile.am	2009-12-17 21:23:04.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for turkish.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = latin3.pl latin5.pl utf8.pl.in
+subdirs += turkish
 
-CLEANFILES = \
-	latin3.raw \
-	latin3-doc.tex \
-	latin5.raw \
-	latin5-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	turkish/latin3-lang.xdy turkish/latin3-test.xdy turkish/latin3.xdy \
+	turkish/latin5-lang.xdy turkish/latin5-test.xdy turkish/latin5.xdy \
+	turkish/utf8-lang.xdy turkish/utf8-test.xdy turkish/utf8.xdy
 
+DOCS_LATIN3 += \
+	turkish/latin3-doc.tex
 
+DOCS_LATIN5 += \
+	turkish/latin5-doc.tex
 
+DOCS_UTF8_LAT += \
+	turkish/utf8-doc.tex
+
+EXTRA_DIST += \
+	turkish/utf8.pl.in
+
+CLEANFILES += \
+	turkish/latin3.pl turkish/latin3.raw \
+	turkish/latin5.pl turkish/latin5.raw \
+	turkish/utf8.pl turkish/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/ukrainian/Makefile.am xindy/make-rules/alphabets/ukrainian/Makefile.am
--- xindy.orig/make-rules/alphabets/ukrainian/Makefile.am	2008-01-09 15:01:10.000000000 +0100
+++ xindy/make-rules/alphabets/ukrainian/Makefile.am	2009-12-17 21:23:41.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for ukrainian.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1251.pl koi8-u.pl utf8.pl.in
+subdirs += ukrainian
 
-CLEANFILES = \
-	cp1251.raw \
-	cp1251-doc.tex \
-	koi8-u.raw \
-	koi8-u-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	ukrainian/cp1251-lang.xdy ukrainian/cp1251-test.xdy ukrainian/cp1251.xdy \
+	ukrainian/koi8-u-lang.xdy ukrainian/koi8-u-test.xdy ukrainian/koi8-u.xdy \
+	ukrainian/utf8-lang.xdy ukrainian/utf8-test.xdy ukrainian/utf8.xdy
 
+DOCS_CYR_1251 += \
+	ukrainian/cp1251-doc.tex
 
+DOCS_CYR_KOI8_U += \
+	ukrainian/koi8-u-doc.tex
 
+DOCS_UTF8_CYR += \
+	ukrainian/utf8-doc.tex
+
+EXTRA_DIST += \
+	ukrainian/utf8.pl.in
+
+CLEANFILES += \
+	ukrainian/cp1251.pl ukrainian/cp1251.raw \
+	ukrainian/koi8-u.pl ukrainian/koi8-u.raw \
+	ukrainian/utf8.pl ukrainian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/upper-sorbian/Makefile.am xindy/make-rules/alphabets/upper-sorbian/Makefile.am
--- xindy.orig/make-rules/alphabets/upper-sorbian/Makefile.am	2008-01-09 15:01:09.000000000 +0100
+++ xindy/make-rules/alphabets/upper-sorbian/Makefile.am	2009-12-17 21:24:19.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for .
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,16 +18,27 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = cp1250.pl latin2.pl utf8.pl.in
+subdirs += upper-sorbian
 
-CLEANFILES = \
-	cp1250.raw \
-	cp1250-doc.tex \
-	latin2.raw \
-	latin2-doc.tex \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	upper-sorbian/cp1250-lang.xdy upper-sorbian/cp1250-test.xdy upper-sorbian/cp1250.xdy \
+	upper-sorbian/latin2-lang.xdy upper-sorbian/latin2-test.xdy upper-sorbian/latin2.xdy \
+	upper-sorbian/utf8-lang.xdy upper-sorbian/utf8-test.xdy upper-sorbian/utf8.xdy
 
+DOCS_CP1250 += \
+	upper-sorbian/cp1250-doc.tex
 
+DOCS_LATIN2 += \
+	upper-sorbian/latin2-doc.tex
 
+DOCS_UTF8_LAT += \
+	upper-sorbian/utf8-doc.tex
+
+EXTRA_DIST += \
+	upper-sorbian/utf8.pl.in
+
+CLEANFILES += \
+	upper-sorbian/cp1250.pl upper-sorbian/cp1250.raw \
+	upper-sorbian/latin2.pl upper-sorbian/latin2.raw \
+	upper-sorbian/utf8.pl upper-sorbian/utf8.raw
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/vietnamese/Makefile.am xindy/make-rules/alphabets/vietnamese/Makefile.am
--- xindy.orig/make-rules/alphabets/vietnamese/Makefile.am	2008-01-09 15:01:05.000000000 +0100
+++ xindy/make-rules/alphabets/vietnamese/Makefile.am	2009-12-17 21:24:35.000000000 +0100
@@ -1,7 +1,6 @@
-## Use `autoreconf' in top-level directory to recreate all
-## autoconf/automake files.
-## Or, call `automake Makefile' to create only Makefile.in.
+## Makefile fragment for vietnamese.
 
+## Copyright (C) 2009 by Peter Breitenlohner.
 ## Copyright (C) 2004-2005 by Gour.
 ##
 ## This program is free software; you can redistribute it and/or
@@ -19,12 +18,17 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-EXTRA_DIST = utf8.pl.in
+subdirs += vietnamese
 
-CLEANFILES = \
-	utf8.raw \
-	utf8-doc.tex
+nobase_nodist_lang_DATA += \
+	vietnamese/utf8-lang.xdy vietnamese/utf8-test.xdy vietnamese/utf8.xdy
 
+DOCS_UTF8_LAT += \
+	vietnamese/utf8-doc.tex
 
+EXTRA_DIST += \
+	vietnamese/utf8.pl.in
 
+CLEANFILES += \
+	vietnamese/utf8.pl vietnamese/utf8.raw
 
